{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Création de la BDD: \n",
    "\n",
    "Utiliser pandas pour séparer les données afin de les mettre dans plusieurs tables d'une base de donnée que vous allez appeler pubmed (en faisant normalisant vos données).\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T10:26:59.346504Z",
     "start_time": "2020-06-24T10:26:58.409037Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T10:32:22.835686Z",
     "start_time": "2020-06-24T10:32:22.763121Z"
    }
   },
   "outputs": [],
   "source": [
    "pubmed_data = pd.read_csv('/Users/Johanna/Documents/SIMPLON DATA IA/IA/ML PROJET/PROJET API SQL/pubmed.csv')\n",
    "pubmed_data = pubmed_data.drop(['Unnamed: 0'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T10:32:25.470900Z",
     "start_time": "2020-06-24T10:32:25.426240Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>publication_date</th>\n",
       "      <th>keywords</th>\n",
       "      <th>journal</th>\n",
       "      <th>doi</th>\n",
       "      <th>last_author</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome-Wide Study of the GATL Gene Family in G...</td>\n",
       "      <td>\\nPectin is a major polysaccharide component t...</td>\n",
       "      <td>31935825</td>\n",
       "      <td>2020-01-16 06:00:00</td>\n",
       "      <td>GhGATL, gene, family, Gossypium, hirsutum, L.,...</td>\n",
       "      <td>Genes, Genes (Basel)</td>\n",
       "      <td>10.3390/genes11010064</td>\n",
       "      <td>Zheng, Lei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Selection and validation of reliable reference...</td>\n",
       "      <td>\\nTolypocladium guangdongense, formerly known ...</td>\n",
       "      <td>31978511</td>\n",
       "      <td>2020-01-25 06:00:00</td>\n",
       "      <td>Cold, and, heat, stress, Cordyceps, guangdonge...</td>\n",
       "      <td>Gene, Gene</td>\n",
       "      <td>10.1016/j.gene.2020.144380</td>\n",
       "      <td>Wang, Gangzheng</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Validation of the Reference Genes for the Gene...</td>\n",
       "      <td>\\nThe selection of a suitable reference gene a...</td>\n",
       "      <td>32235512</td>\n",
       "      <td>2020-04-03 06:00:00</td>\n",
       "      <td>chicken, B, cells, housekeeping, genes, in, vi...</td>\n",
       "      <td>Genes, Genes (Basel)</td>\n",
       "      <td>10.3390/genes11040372</td>\n",
       "      <td>Dunislawska, Aleksandra</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Conventionally used reference genes are not ou...</td>\n",
       "      <td>\\nThe selection of reference genes is essentia...</td>\n",
       "      <td>31138119</td>\n",
       "      <td>2019-05-30 06:00:00</td>\n",
       "      <td>Human, cancer, RT-qPCR, Reference, genes</td>\n",
       "      <td>BMC bioinformatics, BMC Bioinformatics</td>\n",
       "      <td>10.1186/s12859-019-2809-2</td>\n",
       "      <td>Jo, Jihoon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome-Wide Identification and Gene Expression...</td>\n",
       "      <td>\\nAbscisic acid (ABA) plays important roles in...</td>\n",
       "      <td>31226871</td>\n",
       "      <td>2019-06-23 06:00:00</td>\n",
       "      <td>ABA, receptor, BjuPYLs, Brassica, juncea, var....</td>\n",
       "      <td>Genes, Genes (Basel)</td>\n",
       "      <td>10.3390/genes10060470</td>\n",
       "      <td>Cheng, Chunhong</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               title  \\\n",
       "0  Genome-Wide Study of the GATL Gene Family in G...   \n",
       "1  Selection and validation of reliable reference...   \n",
       "2  Validation of the Reference Genes for the Gene...   \n",
       "3  Conventionally used reference genes are not ou...   \n",
       "4  Genome-Wide Identification and Gene Expression...   \n",
       "\n",
       "                                            abstract  pubmed_id  \\\n",
       "0  \\nPectin is a major polysaccharide component t...   31935825   \n",
       "1  \\nTolypocladium guangdongense, formerly known ...   31978511   \n",
       "2  \\nThe selection of a suitable reference gene a...   32235512   \n",
       "3  \\nThe selection of reference genes is essentia...   31138119   \n",
       "4  \\nAbscisic acid (ABA) plays important roles in...   31226871   \n",
       "\n",
       "      publication_date                                           keywords  \\\n",
       "0  2020-01-16 06:00:00  GhGATL, gene, family, Gossypium, hirsutum, L.,...   \n",
       "1  2020-01-25 06:00:00  Cold, and, heat, stress, Cordyceps, guangdonge...   \n",
       "2  2020-04-03 06:00:00  chicken, B, cells, housekeeping, genes, in, vi...   \n",
       "3  2019-05-30 06:00:00           Human, cancer, RT-qPCR, Reference, genes   \n",
       "4  2019-06-23 06:00:00  ABA, receptor, BjuPYLs, Brassica, juncea, var....   \n",
       "\n",
       "                                  journal                         doi  \\\n",
       "0                    Genes, Genes (Basel)       10.3390/genes11010064   \n",
       "1                              Gene, Gene  10.1016/j.gene.2020.144380   \n",
       "2                    Genes, Genes (Basel)       10.3390/genes11040372   \n",
       "3  BMC bioinformatics, BMC Bioinformatics   10.1186/s12859-019-2809-2   \n",
       "4                    Genes, Genes (Basel)       10.3390/genes10060470   \n",
       "\n",
       "               last_author  \n",
       "0               Zheng, Lei  \n",
       "1          Wang, Gangzheng  \n",
       "2  Dunislawska, Aleksandra  \n",
       "3               Jo, Jihoon  \n",
       "4          Cheng, Chunhong  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T12:44:50.546238Z",
     "start_time": "2020-06-24T12:44:50.521370Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CategoricalDtype(categories=['Abramovs, Nikita', 'Acharya, Sudipta', 'Adeola, Fowotade',\n",
       "                  'Adler, Marlen', 'Afgar, Ali', 'Ahmed, Mahmoud',\n",
       "                  'Ahn, Seung-Joon', 'Ai, Ye', 'Akdeniz, Demet',\n",
       "                  'Alam, Intikhab',\n",
       "                  ...\n",
       "                  'Zhu, Ting', 'Zion, Tricia N', 'Zmienko, Agnieszka',\n",
       "                  'Zwaenepoel, Arthur', 'da Costa E Silva Carvalho, Simone',\n",
       "                  'da Silva, Marcel Fernando', 'de Vries, Sophie',\n",
       "                  'dos Santos, Odelta', 'van Parijs, F R D',\n",
       "                  'Łopieńska-Biernat, Elżbieta'],\n",
       "                 ordered=False)"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_data[\"last_author\"] = pubmed_data[\"last_author\"].astype('category')\n",
    "pubmed_data[\"journal\"] = pubmed_data[\"journal\"].astype('category')\n",
    "\n",
    "pubmed_data[\"last_author\"].dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:06:34.405037Z",
     "start_time": "2020-06-24T13:06:34.316609Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>publication_date</th>\n",
       "      <th>keywords</th>\n",
       "      <th>journal</th>\n",
       "      <th>doi</th>\n",
       "      <th>last_author</th>\n",
       "      <th>auteur_id</th>\n",
       "      <th>journal_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome-Wide Study of the GATL Gene Family in G...</td>\n",
       "      <td>\\nPectin is a major polysaccharide component t...</td>\n",
       "      <td>31935825</td>\n",
       "      <td>2020-01-16 06:00:00</td>\n",
       "      <td>GhGATL, gene, family, Gossypium, hirsutum, L.,...</td>\n",
       "      <td>Genes, Genes (Basel)</td>\n",
       "      <td>10.3390/genes11010064</td>\n",
       "      <td>Zheng, Lei</td>\n",
       "      <td>938</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Selection and validation of reliable reference...</td>\n",
       "      <td>\\nTolypocladium guangdongense, formerly known ...</td>\n",
       "      <td>31978511</td>\n",
       "      <td>2020-01-25 06:00:00</td>\n",
       "      <td>Cold, and, heat, stress, Cordyceps, guangdonge...</td>\n",
       "      <td>Gene, Gene</td>\n",
       "      <td>10.1016/j.gene.2020.144380</td>\n",
       "      <td>Wang, Gangzheng</td>\n",
       "      <td>731</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Validation of the Reference Genes for the Gene...</td>\n",
       "      <td>\\nThe selection of a suitable reference gene a...</td>\n",
       "      <td>32235512</td>\n",
       "      <td>2020-04-03 06:00:00</td>\n",
       "      <td>chicken, B, cells, housekeeping, genes, in, vi...</td>\n",
       "      <td>Genes, Genes (Basel)</td>\n",
       "      <td>10.3390/genes11040372</td>\n",
       "      <td>Dunislawska, Aleksandra</td>\n",
       "      <td>148</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Conventionally used reference genes are not ou...</td>\n",
       "      <td>\\nThe selection of reference genes is essentia...</td>\n",
       "      <td>31138119</td>\n",
       "      <td>2019-05-30 06:00:00</td>\n",
       "      <td>Human, cancer, RT-qPCR, Reference, genes</td>\n",
       "      <td>BMC bioinformatics, BMC Bioinformatics</td>\n",
       "      <td>10.1186/s12859-019-2809-2</td>\n",
       "      <td>Jo, Jihoon</td>\n",
       "      <td>314</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome-Wide Identification and Gene Expression...</td>\n",
       "      <td>\\nAbscisic acid (ABA) plays important roles in...</td>\n",
       "      <td>31226871</td>\n",
       "      <td>2019-06-23 06:00:00</td>\n",
       "      <td>ABA, receptor, BjuPYLs, Brassica, juncea, var....</td>\n",
       "      <td>Genes, Genes (Basel)</td>\n",
       "      <td>10.3390/genes10060470</td>\n",
       "      <td>Cheng, Chunhong</td>\n",
       "      <td>97</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               title  \\\n",
       "0  Genome-Wide Study of the GATL Gene Family in G...   \n",
       "1  Selection and validation of reliable reference...   \n",
       "2  Validation of the Reference Genes for the Gene...   \n",
       "3  Conventionally used reference genes are not ou...   \n",
       "4  Genome-Wide Identification and Gene Expression...   \n",
       "\n",
       "                                            abstract  pubmed_id  \\\n",
       "0  \\nPectin is a major polysaccharide component t...   31935825   \n",
       "1  \\nTolypocladium guangdongense, formerly known ...   31978511   \n",
       "2  \\nThe selection of a suitable reference gene a...   32235512   \n",
       "3  \\nThe selection of reference genes is essentia...   31138119   \n",
       "4  \\nAbscisic acid (ABA) plays important roles in...   31226871   \n",
       "\n",
       "      publication_date                                           keywords  \\\n",
       "0  2020-01-16 06:00:00  GhGATL, gene, family, Gossypium, hirsutum, L.,...   \n",
       "1  2020-01-25 06:00:00  Cold, and, heat, stress, Cordyceps, guangdonge...   \n",
       "2  2020-04-03 06:00:00  chicken, B, cells, housekeeping, genes, in, vi...   \n",
       "3  2019-05-30 06:00:00           Human, cancer, RT-qPCR, Reference, genes   \n",
       "4  2019-06-23 06:00:00  ABA, receptor, BjuPYLs, Brassica, juncea, var....   \n",
       "\n",
       "                                  journal                         doi  \\\n",
       "0                    Genes, Genes (Basel)       10.3390/genes11010064   \n",
       "1                              Gene, Gene  10.1016/j.gene.2020.144380   \n",
       "2                    Genes, Genes (Basel)       10.3390/genes11040372   \n",
       "3  BMC bioinformatics, BMC Bioinformatics   10.1186/s12859-019-2809-2   \n",
       "4                    Genes, Genes (Basel)       10.3390/genes10060470   \n",
       "\n",
       "               last_author  auteur_id  journal_id  \n",
       "0               Zheng, Lei        938         140  \n",
       "1          Wang, Gangzheng        731         136  \n",
       "2  Dunislawska, Aleksandra        148         140  \n",
       "3               Jo, Jihoon        314          27  \n",
       "4          Cheng, Chunhong         97         140  "
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_data[\"auteur_id\"] = pubmed_data[\"last_author\"].cat.codes\n",
    "pubmed_data[\"journal_id\"] = pubmed_data[\"journal\"].cat.codes\n",
    "pubmed_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T10:35:29.218357Z",
     "start_time": "2020-06-24T10:35:29.188733Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene, Gene                                                        102\n",
       "PloS one, PLoS ONE                                                 75\n",
       "Genes, Genes (Basel)                                               75\n",
       "Scientific reports, Sci Rep                                        44\n",
       "BMC genomics, BMC Genomics                                         30\n",
       "                                                                 ... \n",
       "Animal : an international journal of animal bioscience, Animal      1\n",
       "Ethiopian journal of health sciences, Ethiop J Health Sci           1\n",
       "Methods (San Diego, Calif.), Methods                                1\n",
       "Archives of oral biology, Arch. Oral Biol.                          1\n",
       "Bioresource technology, Bioresour. Technol.                         1\n",
       "Name: journal, Length: 356, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_data['journal'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T12:57:25.563917Z",
     "start_time": "2020-06-24T12:57:25.547261Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "356"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pubmed_data['journal'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:02:10.517931Z",
     "start_time": "2020-06-24T13:02:10.462184Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(356, 2)"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "journal_order = pd.DataFrame(pubmed_data[[\"journal\",\"journal_id\"]])\n",
    "journal_order = journal_order.sort_values(by=['journal_id'])\n",
    "journal_unique = journal_order.drop_duplicates(subset=[\"journal\",\"journal_id\"])\n",
    "journal_unique.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T10:35:50.176438Z",
     "start_time": "2020-06-24T10:35:50.142113Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "He, Wenwu               3\n",
       "Lu, Wenzong             3\n",
       "Kong, Qiusheng          3\n",
       "Wang, Zheng             3\n",
       "Wang, Ying              2\n",
       "                       ..\n",
       "Bao, Yongbo             1\n",
       "Vuillaume, M-L          1\n",
       "Wu, K                   1\n",
       "Pourbagher, Roghayeh    1\n",
       "Lachmann, Alexander     1\n",
       "Name: last_author, Length: 968, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_data['last_author'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T12:21:27.793725Z",
     "start_time": "2020-06-24T12:21:27.764124Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>auteur</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Zheng, Lei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Wang, Gangzheng</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Dunislawska, Aleksandra</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Jo, Jihoon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Cheng, Chunhong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>963</th>\n",
       "      <td>You, Yanchun</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>Reynolds, David L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>Iverson, Eric A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966</th>\n",
       "      <td>Maeda, Kiyohiro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>967</th>\n",
       "      <td>Xiao, Yun</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>968 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      auteur\n",
       "0                 Zheng, Lei\n",
       "1            Wang, Gangzheng\n",
       "2    Dunislawska, Aleksandra\n",
       "3                 Jo, Jihoon\n",
       "4            Cheng, Chunhong\n",
       "..                       ...\n",
       "963             You, Yanchun\n",
       "964        Reynolds, David L\n",
       "965          Iverson, Eric A\n",
       "966          Maeda, Kiyohiro\n",
       "967                Xiao, Yun\n",
       "\n",
       "[968 rows x 1 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "author_unique = list(pubmed_data['last_author'].unique())\n",
    "auteur_df = pd.DataFrame(author_unique, columns=['auteur'])\n",
    "auteur_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:08:48.190641Z",
     "start_time": "2020-06-24T13:08:48.127210Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>last_author</th>\n",
       "      <th>auteur_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>Abramovs, Nikita</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>Acharya, Sudipta</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>753</th>\n",
       "      <td>Adeola, Fowotade</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>733</th>\n",
       "      <td>Adler, Marlen</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>Afgar, Ali</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>da Silva, Marcel Fernando</td>\n",
       "      <td>963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>de Vries, Sophie</td>\n",
       "      <td>964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>dos Santos, Odelta</td>\n",
       "      <td>965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>van Parijs, F R D</td>\n",
       "      <td>966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>Łopieńska-Biernat, Elżbieta</td>\n",
       "      <td>967</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>968 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     last_author  auteur_id\n",
       "594             Abramovs, Nikita          0\n",
       "392             Acharya, Sudipta          1\n",
       "753             Adeola, Fowotade          2\n",
       "733                Adler, Marlen          3\n",
       "572                   Afgar, Ali          4\n",
       "..                           ...        ...\n",
       "649    da Silva, Marcel Fernando        963\n",
       "490             de Vries, Sophie        964\n",
       "792           dos Santos, Odelta        965\n",
       "851            van Parijs, F R D        966\n",
       "463  Łopieńska-Biernat, Elżbieta        967\n",
       "\n",
       "[968 rows x 2 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "author_order = pd.DataFrame(pubmed_data[[\"last_author\",\"auteur_id\"]])\n",
    "author_order = author_order.sort_values(by=['auteur_id'])\n",
    "author_unique = author_order.drop_duplicates(subset=[\"last_author\",\"auteur_id\"])\n",
    "author_unique.shape\n",
    "author_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T10:36:33.287198Z",
     "start_time": "2020-06-24T10:36:33.243760Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2019-04-27 06:00:00    8\n",
       "2020-03-19 06:00:00    5\n",
       "2020-01-01 06:00:00    5\n",
       "2017-10-27 06:00:00    5\n",
       "2019-09-26 06:00:00    4\n",
       "                      ..\n",
       "2020-01-16 06:00:00    1\n",
       "2017-06-22 06:00:00    1\n",
       "2015-07-05 06:00:00    1\n",
       "2016-10-06 06:00:00    1\n",
       "2016-10-12 06:00:00    1\n",
       "Name: publication_date, Length: 725, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed_data['publication_date'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:09:00.193484Z",
     "start_time": "2020-06-24T13:09:00.112789Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>last_author</th>\n",
       "      <th>auteur_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>Abramovs, Nikita</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>Acharya, Sudipta</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>753</th>\n",
       "      <td>Adeola, Fowotade</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>733</th>\n",
       "      <td>Adler, Marlen</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>Afgar, Ali</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>da Silva, Marcel Fernando</td>\n",
       "      <td>963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>de Vries, Sophie</td>\n",
       "      <td>964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>dos Santos, Odelta</td>\n",
       "      <td>965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>van Parijs, F R D</td>\n",
       "      <td>966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>Łopieńska-Biernat, Elżbieta</td>\n",
       "      <td>967</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>968 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     last_author  auteur_id\n",
       "594             Abramovs, Nikita          0\n",
       "392             Acharya, Sudipta          1\n",
       "753             Adeola, Fowotade          2\n",
       "733                Adler, Marlen          3\n",
       "572                   Afgar, Ali          4\n",
       "..                           ...        ...\n",
       "649    da Silva, Marcel Fernando        963\n",
       "490             de Vries, Sophie        964\n",
       "792           dos Santos, Odelta        965\n",
       "851            van Parijs, F R D        966\n",
       "463  Łopieńska-Biernat, Elżbieta        967\n",
       "\n",
       "[968 rows x 2 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split column\n",
    "author_df = pd.DataFrame(author_unique)\n",
    "author_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:09:47.384025Z",
     "start_time": "2020-06-24T13:09:47.241080Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>doi</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>keywords</th>\n",
       "      <th>publication_date</th>\n",
       "      <th>auteur_id</th>\n",
       "      <th>journal_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>31935825</td>\n",
       "      <td>10.3390/genes11010064</td>\n",
       "      <td>Genome-Wide Study of the GATL Gene Family in G...</td>\n",
       "      <td>\\nPectin is a major polysaccharide component t...</td>\n",
       "      <td>GhGATL, gene, family, Gossypium, hirsutum, L.,...</td>\n",
       "      <td>2020-01-16 06:00:00</td>\n",
       "      <td>938</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>31978511</td>\n",
       "      <td>10.1016/j.gene.2020.144380</td>\n",
       "      <td>Selection and validation of reliable reference...</td>\n",
       "      <td>\\nTolypocladium guangdongense, formerly known ...</td>\n",
       "      <td>Cold, and, heat, stress, Cordyceps, guangdonge...</td>\n",
       "      <td>2020-01-25 06:00:00</td>\n",
       "      <td>731</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>32235512</td>\n",
       "      <td>10.3390/genes11040372</td>\n",
       "      <td>Validation of the Reference Genes for the Gene...</td>\n",
       "      <td>\\nThe selection of a suitable reference gene a...</td>\n",
       "      <td>chicken, B, cells, housekeeping, genes, in, vi...</td>\n",
       "      <td>2020-04-03 06:00:00</td>\n",
       "      <td>148</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>31138119</td>\n",
       "      <td>10.1186/s12859-019-2809-2</td>\n",
       "      <td>Conventionally used reference genes are not ou...</td>\n",
       "      <td>\\nThe selection of reference genes is essentia...</td>\n",
       "      <td>Human, cancer, RT-qPCR, Reference, genes</td>\n",
       "      <td>2019-05-30 06:00:00</td>\n",
       "      <td>314</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>31226871</td>\n",
       "      <td>10.3390/genes10060470</td>\n",
       "      <td>Genome-Wide Identification and Gene Expression...</td>\n",
       "      <td>\\nAbscisic acid (ABA) plays important roles in...</td>\n",
       "      <td>ABA, receptor, BjuPYLs, Brassica, juncea, var....</td>\n",
       "      <td>2019-06-23 06:00:00</td>\n",
       "      <td>97</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>29620473</td>\n",
       "      <td>10.1139/gen-2017-0176</td>\n",
       "      <td>Selecting and validating reference genes for q...</td>\n",
       "      <td>\\nGene expression analysis provides important ...</td>\n",
       "      <td>diamondback, moth, expression, pattern, gènes,...</td>\n",
       "      <td>2018-04-06 06:00:00</td>\n",
       "      <td>859</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>27125778</td>\n",
       "      <td>10.1111/mmi.13408</td>\n",
       "      <td>Base J represses genes at the end of polycistr...</td>\n",
       "      <td>\\nThe genomes of kinetoplastids are organized ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016-04-30 06:00:00</td>\n",
       "      <td>593</td>\n",
       "      <td>262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>28148789</td>\n",
       "      <td>10.1128/JVI.02406-16</td>\n",
       "      <td>Extreme Mutation Tolerance: Nearly Half of the...</td>\n",
       "      <td>\\nViruses infecting the Archaea harbor a treme...</td>\n",
       "      <td>Archaea, morphogenesis, mutational, studies, v...</td>\n",
       "      <td>2017-02-06 06:00:00</td>\n",
       "      <td>299</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>30797967</td>\n",
       "      <td>10.1016/j.biosystems.2019.02.008</td>\n",
       "      <td>Gene grouping strategy for network modeling fr...</td>\n",
       "      <td>\\nSeveral algorithms have been proposed for mo...</td>\n",
       "      <td>Bayesian, network, Gene, grouping, Gene, regul...</td>\n",
       "      <td>2019-02-25 06:00:00</td>\n",
       "      <td>467</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>26039571</td>\n",
       "      <td>10.1038/srep10889</td>\n",
       "      <td>Gene Perturbation Atlas (GPA): a single-gene p...</td>\n",
       "      <td>\\nGenome-wide transcriptome profiling after ge...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-06-04 06:00:00</td>\n",
       "      <td>804</td>\n",
       "      <td>319</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     pubmed_id                               doi  \\\n",
       "0     31935825             10.3390/genes11010064   \n",
       "1     31978511        10.1016/j.gene.2020.144380   \n",
       "2     32235512             10.3390/genes11040372   \n",
       "3     31138119         10.1186/s12859-019-2809-2   \n",
       "4     31226871             10.3390/genes10060470   \n",
       "..         ...                               ...   \n",
       "995   29620473             10.1139/gen-2017-0176   \n",
       "996   27125778                 10.1111/mmi.13408   \n",
       "997   28148789              10.1128/JVI.02406-16   \n",
       "998   30797967  10.1016/j.biosystems.2019.02.008   \n",
       "999   26039571                 10.1038/srep10889   \n",
       "\n",
       "                                                 title  \\\n",
       "0    Genome-Wide Study of the GATL Gene Family in G...   \n",
       "1    Selection and validation of reliable reference...   \n",
       "2    Validation of the Reference Genes for the Gene...   \n",
       "3    Conventionally used reference genes are not ou...   \n",
       "4    Genome-Wide Identification and Gene Expression...   \n",
       "..                                                 ...   \n",
       "995  Selecting and validating reference genes for q...   \n",
       "996  Base J represses genes at the end of polycistr...   \n",
       "997  Extreme Mutation Tolerance: Nearly Half of the...   \n",
       "998  Gene grouping strategy for network modeling fr...   \n",
       "999  Gene Perturbation Atlas (GPA): a single-gene p...   \n",
       "\n",
       "                                              abstract  \\\n",
       "0    \\nPectin is a major polysaccharide component t...   \n",
       "1    \\nTolypocladium guangdongense, formerly known ...   \n",
       "2    \\nThe selection of a suitable reference gene a...   \n",
       "3    \\nThe selection of reference genes is essentia...   \n",
       "4    \\nAbscisic acid (ABA) plays important roles in...   \n",
       "..                                                 ...   \n",
       "995  \\nGene expression analysis provides important ...   \n",
       "996  \\nThe genomes of kinetoplastids are organized ...   \n",
       "997  \\nViruses infecting the Archaea harbor a treme...   \n",
       "998  \\nSeveral algorithms have been proposed for mo...   \n",
       "999  \\nGenome-wide transcriptome profiling after ge...   \n",
       "\n",
       "                                              keywords     publication_date  \\\n",
       "0    GhGATL, gene, family, Gossypium, hirsutum, L.,...  2020-01-16 06:00:00   \n",
       "1    Cold, and, heat, stress, Cordyceps, guangdonge...  2020-01-25 06:00:00   \n",
       "2    chicken, B, cells, housekeeping, genes, in, vi...  2020-04-03 06:00:00   \n",
       "3             Human, cancer, RT-qPCR, Reference, genes  2019-05-30 06:00:00   \n",
       "4    ABA, receptor, BjuPYLs, Brassica, juncea, var....  2019-06-23 06:00:00   \n",
       "..                                                 ...                  ...   \n",
       "995  diamondback, moth, expression, pattern, gènes,...  2018-04-06 06:00:00   \n",
       "996                                                NaN  2016-04-30 06:00:00   \n",
       "997  Archaea, morphogenesis, mutational, studies, v...  2017-02-06 06:00:00   \n",
       "998  Bayesian, network, Gene, grouping, Gene, regul...  2019-02-25 06:00:00   \n",
       "999                                                NaN  2015-06-04 06:00:00   \n",
       "\n",
       "     auteur_id  journal_id  \n",
       "0          938         140  \n",
       "1          731         136  \n",
       "2          148         140  \n",
       "3          314          27  \n",
       "4           97         140  \n",
       "..         ...         ...  \n",
       "995        859         150  \n",
       "996        593         262  \n",
       "997        299         229  \n",
       "998        467          47  \n",
       "999        804         319  \n",
       "\n",
       "[1000 rows x 8 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "publication_df = pd.DataFrame(pubmed_data[['pubmed_id','doi','title','abstract','keywords','publication_date','auteur_id','journal_id']])\n",
    "publication_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:03:07.033877Z",
     "start_time": "2020-06-24T13:03:06.978291Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>journal</th>\n",
       "      <th>journal_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>3 Biotech, 3 Biotech</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>Acta parasitologica, Acta Parasitol.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>Acta reumatologica portuguesa, Acta Reumatol Port</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>Acta veterinaria Hungarica, Acta Vet. Hung.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>Advances in clinical and experimental medicine...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705</th>\n",
       "      <td>World journal of surgical oncology, World J Su...</td>\n",
       "      <td>351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>Ying yong sheng tai xue bao = The journal of a...</td>\n",
       "      <td>352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>Yonsei medical journal, Yonsei Med. J.</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>766</th>\n",
       "      <td>Zeitschrift fur Naturforschung. C, Journal of ...</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>917</th>\n",
       "      <td>Zoological science, Zool. Sci.</td>\n",
       "      <td>355</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>356 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               journal  journal_id\n",
       "320                               3 Biotech, 3 Biotech           0\n",
       "463               Acta parasitologica, Acta Parasitol.           1\n",
       "536  Acta reumatologica portuguesa, Acta Reumatol Port           2\n",
       "871        Acta veterinaria Hungarica, Acta Vet. Hung.           3\n",
       "992  Advances in clinical and experimental medicine...           4\n",
       "..                                                 ...         ...\n",
       "705  World journal of surgical oncology, World J Su...         351\n",
       "298  Ying yong sheng tai xue bao = The journal of a...         352\n",
       "217             Yonsei medical journal, Yonsei Med. J.         353\n",
       "766  Zeitschrift fur Naturforschung. C, Journal of ...         354\n",
       "917                     Zoological science, Zool. Sci.         355\n",
       "\n",
       "[356 rows x 2 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "journal_df = pd.DataFrame(journal_unique, columns=[\"journal\",\"journal_id\"])\n",
    "journal_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T14:16:58.564017Z",
     "start_time": "2020-06-24T14:16:58.487556Z"
    }
   },
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "\n",
    "conn = sqlite3.connect('pubmed.db')  # You can create a new database by changing the name within the quotes\n",
    "c = conn.cursor() # The database will be saved in the location where your 'py' file is saved\n",
    "\n",
    "# Create table - PUBLICATION\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS PUBLICATION\n",
    "             ([pubmed_id] INTEGER PRIMARY KEY,[doi] text, [title] text, [abstract] text, [keywords] text, [publication_date] date, [auteur_id] INTEGER NOT NULL, [journal_id] INTEGER NOT NULL, \n",
    "             FOREIGN KEY (auteur_id) REFERENCES AUTEUR (auteur_id),\n",
    "             FOREIGN KEY (journal_id) REFERENCES JOURNAL (journal_id))''')\n",
    "          \n",
    "# Create table - AUTEUR\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS AUTEUR\n",
    "             ([auteur_id] INTEGER PRIMARY KEY,[auteur] text)''')\n",
    "        \n",
    "# Create table - JOURNAL_TABLE\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS JOURNAL\n",
    "             ([journal_id] INTEGER PRIMARY KEY, [journal] text)''')\n",
    "                 \n",
    "conn.commit()\n",
    "\n",
    "# Note that the syntax to create new tables should only be used once in the code (unless you dropped the table/s at the end of the code). \n",
    "# The [generated_id] column is used to set an auto-increment ID for each record\n",
    "# When creating a new table, you can add both the field names as well as the field formats (e.g., Text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T14:16:59.882299Z",
     "start_time": "2020-06-24T14:16:59.558399Z"
    }
   },
   "outputs": [],
   "source": [
    "publication_df.to_sql('PUBLICATION', con=conn, if_exists='append', index = False)\n",
    "# Insert the values from the dataframe into the table 'PUBLICATION'\n",
    "\n",
    "auteur_df.to_sql('AUTEUR', con=conn, if_exists='append', index = False)\n",
    "journal_df.to_sql('JOURNAL_TABLE', con=conn, if_exists='append', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T14:17:01.089396Z",
     "start_time": "2020-06-24T14:17:00.839078Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('3 Biotech, 3 Biotech', 0),\n",
       " ('Acta parasitologica, Acta Parasitol.', 1),\n",
       " ('Acta reumatologica portuguesa, Acta Reumatol Port', 2),\n",
       " ('Acta veterinaria Hungarica, Acta Vet. Hung.', 3),\n",
       " ('Advances in clinical and experimental medicine : official organ Wroclaw Medical University, Adv Clin Exp Med',\n",
       "  4),\n",
       " ('Aging clinical and experimental research, Aging Clin Exp Res', 5),\n",
       " ('Aging, Aging (Albany NY)', 6),\n",
       " ('American journal of human genetics, Am. J. Hum. Genet.', 7),\n",
       " ('American journal of medical genetics. Part B, Neuropsychiatric genetics : the official publication of the International Society of Psychiatric Genetics, Am. J. Med. Genet. B Neuropsychiatr. Genet.',\n",
       "  8),\n",
       " ('Anaerobe, Anaerobe', 9),\n",
       " ('Andrologia, Andrologia', 10),\n",
       " ('Animal : an international journal of animal bioscience, Animal', 11),\n",
       " ('Annals of anatomy = Anatomischer Anzeiger : official organ of the Anatomische Gesellschaft, Ann. Anat.',\n",
       "  12),\n",
       " ('Annals of translational medicine, Ann Transl Med', 13),\n",
       " ('Antonie van Leeuwenhoek, Antonie Van Leeuwenhoek', 14),\n",
       " ('Applied and environmental microbiology, Appl. Environ. Microbiol.', 15),\n",
       " ('Applied biochemistry and biotechnology, Appl. Biochem. Biotechnol.', 16),\n",
       " ('Applied microbiology and biotechnology, Appl. Microbiol. Biotechnol.', 17),\n",
       " ('Aquatic toxicology (Amsterdam, Netherlands), Aquat. Toxicol.', 18),\n",
       " ('Archives of dermatological research, Arch. Dermatol. Res.', 19),\n",
       " ('Archives of gynecology and obstetrics, Arch. Gynecol. Obstet.', 20),\n",
       " ('Archives of insect biochemistry and physiology, Arch. Insect Biochem. Physiol.',\n",
       "  21),\n",
       " ('Archives of oral biology, Arch. Oral Biol.', 22),\n",
       " ('Archives of virology, Arch. Virol.', 23),\n",
       " ('Asia-Pacific journal of ophthalmology (Philadelphia, Pa.), Asia Pac J Ophthalmol (Phila)',\n",
       "  24),\n",
       " ('Asia-Pacific psychiatry : official journal of the Pacific Rim College of Psychiatrists, Asia Pac Psychiatry',\n",
       "  25),\n",
       " ('BMB reports, BMB Rep', 26),\n",
       " ('BMC bioinformatics, BMC Bioinformatics', 27),\n",
       " ('BMC biology, BMC Biol.', 28),\n",
       " ('BMC biotechnology, BMC Biotechnol.', 29),\n",
       " ('BMC cancer, BMC Cancer', 30),\n",
       " ('BMC cardiovascular disorders, BMC Cardiovasc Disord', 31),\n",
       " ('BMC evolutionary biology, BMC Evol. Biol.', 32),\n",
       " ('BMC genetics, BMC Genet.', 33),\n",
       " ('BMC genomics, BMC Genomics', 34),\n",
       " ('BMC immunology, BMC Immunol.', 35),\n",
       " ('BMC medical ethics, BMC Med Ethics', 36),\n",
       " ('BMC medical genetics, BMC Med. Genet.', 37),\n",
       " ('BMC medical genomics, BMC Med Genomics', 38),\n",
       " ('BMC microbiology, BMC Microbiol.', 39),\n",
       " ('BMC molecular biology, BMC Mol. Biol.', 40),\n",
       " ('BMC neuroscience, BMC Neurosci', 41),\n",
       " ('BMC plant biology, BMC Plant Biol.', 42),\n",
       " ('BMC research notes, BMC Res Notes', 43),\n",
       " ('BMC systems biology, BMC Syst Biol', 44),\n",
       " ('Behavioural brain research, Behav. Brain Res.', 45),\n",
       " ('Beijing da xue xue bao. Yi xue ban = Journal of Peking University. Health sciences, Beijing Da Xue Xue Bao',\n",
       "  46),\n",
       " ('Bio Systems, BioSystems', 47),\n",
       " ('BioMed research international, Biomed Res Int', 48),\n",
       " ('Biochemical genetics, Biochem. Genet.', 49),\n",
       " ('Biochimica et biophysica acta. Molecular basis of disease, Biochim Biophys Acta Mol Basis Dis',\n",
       "  50),\n",
       " ('Bioinformatics (Oxford, England), Bioinformatics', 51),\n",
       " ('Biological research, Biol. Res.', 52),\n",
       " ('Biological trace element research, Biol Trace Elem Res', 53),\n",
       " ('Biology methods & protocols, Biol Methods Protoc', 54),\n",
       " ('Biomedical reports, Biomed Rep', 55),\n",
       " ('Biomedicine & pharmacotherapy = Biomedecine & pharmacotherapie, Biomed. Pharmacother.',\n",
       "  56),\n",
       " ('Biometals : an international journal on the role of metal ions in biology, biochemistry, and medicine, Biometals',\n",
       "  57),\n",
       " ('Biomolecules, Biomolecules', 58),\n",
       " ('Bioresource technology, Bioresour. Technol.', 59),\n",
       " ('Bioscience reports, Biosci. Rep.', 60),\n",
       " ('Biotechnology and applied biochemistry, Biotechnol. Appl. Biochem.', 61),\n",
       " ('Biotechnology for biofuels, Biotechnol Biofuels', 62),\n",
       " ('Biotechnology letters, Biotechnol. Lett.', 63),\n",
       " ('Bone, Bone', 64),\n",
       " ('Brain research bulletin, Brain Res. Bull.', 65),\n",
       " ('Brazilian journal of biology = Revista brasleira de biologia, Braz J Biol',\n",
       "  66),\n",
       " ('Breast cancer research and treatment, Breast Cancer Res. Treat.', 67),\n",
       " ('Briefings in functional genomics, Brief Funct Genomics', 68),\n",
       " ('Bulletin of entomological research, Bull. Entomol. Res.', 69),\n",
       " ('Canadian journal of microbiology, Can. J. Microbiol.', 70),\n",
       " ('Cancer biology & therapy, Cancer Biol. Ther.', 71),\n",
       " ('Cancer biomarkers : section A of Disease markers, Cancer Biomark', 72),\n",
       " ('Cancer gene therapy, Cancer Gene Ther.', 73),\n",
       " ('Cancer genetics, Cancer Genet', 74),\n",
       " ('Cancer informatics, Cancer Inform', 75),\n",
       " ('Cancer investigation, Cancer Invest.', 76),\n",
       " ('Cancer management and research, Cancer Manag Res', 77),\n",
       " ('Cancer medicine, Cancer Med', 78),\n",
       " ('Cardiovascular research, Cardiovasc. Res.', 79),\n",
       " ('Cell biochemistry and biophysics, Cell Biochem. Biophys.', 80),\n",
       " ('Cell cycle (Georgetown, Tex.), Cell Cycle', 81),\n",
       " ('Cell journal, Cell J', 82),\n",
       " ('Cell reports, Cell Rep', 83),\n",
       " ('Cells, Cells', 84),\n",
       " ('Cellular oncology (Dordrecht), Cell Oncol (Dordr)', 85),\n",
       " ('Cellular physiology and biochemistry : international journal of experimental cellular physiology, biochemistry, and pharmacology, Cell. Physiol. Biochem.',\n",
       "  86),\n",
       " ('Chromosoma, Chromosoma', 87),\n",
       " ('Chronobiology international, Chronobiol. Int.', 88),\n",
       " ('Circulation, Circulation', 89),\n",
       " ('Clinical and translational science, Clin Transl Sci', 90),\n",
       " ('Computational and mathematical methods in medicine, Comput Math Methods Med',\n",
       "  91),\n",
       " ('Computational and structural biotechnology journal, Comput Struct Biotechnol J',\n",
       "  92),\n",
       " ('Computational biology and chemistry, Comput Biol Chem', 93),\n",
       " ('Computers in biology and medicine, Comput. Biol. Med.', 94),\n",
       " ('Critical reviews in eukaryotic gene expression, Crit. Rev. Eukaryot. Gene Expr.',\n",
       "  95),\n",
       " ('Current genetics, Curr. Genet.', 96),\n",
       " ('Current molecular pharmacology, Curr Mol Pharmacol', 97),\n",
       " ('Cytokine, Cytokine', 98),\n",
       " ('DNA research : an international journal for rapid publication of reports on genes and genomes, DNA Res.',\n",
       "  99),\n",
       " ('Data in brief, Data Brief', 100),\n",
       " ('Developmental biology, Dev. Biol.', 101),\n",
       " ('Discovery medicine, Discov Med', 102),\n",
       " ('Disease models & mechanisms, Dis Model Mech', 103),\n",
       " ('Ecotoxicology and environmental safety, Ecotoxicol. Environ. Saf.', 104),\n",
       " ('Endocrine journal, Endocr. J.', 105),\n",
       " ('Endocrine, Endocrine', 106),\n",
       " ('Environmental microbiology, Environ. Microbiol.', 107),\n",
       " ('Environmental pollution (Barking, Essex : 1987), Environ. Pollut.', 108),\n",
       " ('Environmental science & technology, Environ. Sci. Technol.', 109),\n",
       " ('Environmental toxicology and pharmacology, Environ. Toxicol. Pharmacol.',\n",
       "  110),\n",
       " ('Enzyme and microbial technology, Enzyme Microb. Technol.', 111),\n",
       " ('Epigenetics & chromatin, Epigenetics Chromatin', 112),\n",
       " ('Ethiopian journal of health sciences, Ethiop J Health Sci', 113),\n",
       " ('Eukaryotic cell, Eukaryotic Cell', 114),\n",
       " ('European journal of cancer prevention : the official journal of the European Cancer Prevention Organisation (ECP), Eur. J. Cancer Prev.',\n",
       "  115),\n",
       " ('Experimental and therapeutic medicine, Exp Ther Med', 116),\n",
       " ('Experimental physiology, Exp. Physiol.', 117),\n",
       " ('FEBS open bio, FEBS Open Bio', 118),\n",
       " ('FEMS microbiology letters, FEMS Microbiol. Lett.', 119),\n",
       " ('Fish & shellfish immunology, Fish Shellfish Immunol.', 120),\n",
       " ('Fish physiology and biochemistry, Fish Physiol. Biochem.', 121),\n",
       " ('Frontiers in bioengineering and biotechnology, Front Bioeng Biotechnol',\n",
       "  122),\n",
       " ('Frontiers in genetics, Front Genet', 123),\n",
       " ('Frontiers in immunology, Front Immunol', 124),\n",
       " ('Frontiers in medicine, Front Med (Lausanne)', 125),\n",
       " ('Frontiers in microbiology, Front Microbiol', 126),\n",
       " ('Frontiers in oncology, Front Oncol', 127),\n",
       " ('Frontiers in physiology, Front Physiol', 128),\n",
       " ('Frontiers in plant science, Front Plant Sci', 129),\n",
       " ('Frontiers in systems neuroscience, Front Syst Neurosci', 130),\n",
       " ('Functional & integrative genomics, Funct. Integr. Genomics', 131),\n",
       " ('Fungal genetics and biology : FG & B, Fungal Genet. Biol.', 132),\n",
       " ('G3 (Bethesda, Md.), G3 (Bethesda)', 133),\n",
       " ('Gene expression patterns : GEP, Gene Expr. Patterns', 134),\n",
       " ('Gene reports, Gene Rep', 135),\n",
       " ('Gene, Gene', 136),\n",
       " ('Genes & genetic systems, Genes Genet. Syst.', 137),\n",
       " ('Genes & genomics, Genes Genomics', 138),\n",
       " ('Genes and immunity, Genes Immun.', 139),\n",
       " ('Genes, Genes (Basel)', 140),\n",
       " ('Genetic epidemiology, Genet. Epidemiol.', 141),\n",
       " ('Genetic testing and molecular biomarkers, Genet Test Mol Biomarkers', 142),\n",
       " ('Genetics and molecular research : GMR, Genet. Mol. Res.', 143),\n",
       " ('Genetics, Genetics', 144),\n",
       " ('Genetics, selection, evolution : GSE, Genet. Sel. Evol.', 145),\n",
       " ('Genome biology and evolution, Genome Biol Evol', 146),\n",
       " ('Genome biology, Genome Biol.', 147),\n",
       " ('Genome medicine, Genome Med', 148),\n",
       " ('Genome research, Genome Res.', 149),\n",
       " ('Genome, Genome', 150),\n",
       " ('GigaScience, Gigascience', 151),\n",
       " ('Glycobiology, Glycobiology', 152),\n",
       " ('Hematology (Amsterdam, Netherlands), Hematology', 153),\n",
       " ('Hepatology research : the official journal of the Japan Society of Hepatology, Hepatol. Res.',\n",
       "  154),\n",
       " ('Hereditas, Hereditas', 155),\n",
       " ('Histopathology, Histopathology', 156),\n",
       " ('Huan jing ke xue= Huanjing kexue, Huan Jing Ke Xue', 157),\n",
       " ('Human genetics, Hum. Genet.', 158),\n",
       " ('Human molecular genetics, Hum. Mol. Genet.', 159),\n",
       " ('Human mutation, Hum. Mutat.', 160),\n",
       " ('IEEE/ACM transactions on computational biology and bioinformatics, IEEE/ACM Trans Comput Biol Bioinform',\n",
       "  161),\n",
       " ('IET systems biology, IET Syst Biol', 162),\n",
       " ('Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases, Infect. Genet. Evol.',\n",
       "  163),\n",
       " ('Inflammatory bowel diseases, Inflamm. Bowel Dis.', 164),\n",
       " ('Insect biochemistry and molecular biology, Insect Biochem. Mol. Biol.',\n",
       "  165),\n",
       " ('Insect molecular biology, Insect Mol. Biol.', 166),\n",
       " ('Insects, Insects', 167),\n",
       " ('Interdisciplinary sciences, computational life sciences, Interdiscip Sci',\n",
       "  168),\n",
       " ('International journal for parasitology, Int. J. Parasitol.', 169),\n",
       " ('International journal of antimicrobial agents, Int. J. Antimicrob. Agents',\n",
       "  170),\n",
       " ('International journal of cancer, Int. J. Cancer', 171),\n",
       " ('International journal of food microbiology, Int. J. Food Microbiol.', 172),\n",
       " ('International journal of genomics, Int J Genomics', 173),\n",
       " ('International journal of immunogenetics, Int. J. Immunogenet.', 174),\n",
       " ('International journal of legal medicine, Int. J. Legal Med.', 175),\n",
       " ('International journal of medicinal mushrooms, Int J Med Mushrooms', 176),\n",
       " ('International journal of molecular epidemiology and genetics, Int J Mol Epidemiol Genet',\n",
       "  177),\n",
       " ('International journal of molecular medicine, Int. J. Mol. Med.', 178),\n",
       " ('International journal of molecular sciences, Int J Mol Sci', 179),\n",
       " ('International journal of obesity (2005), Int J Obes (Lond)', 180),\n",
       " ('International journal of ophthalmology, Int J Ophthalmol', 181),\n",
       " ('International journal of rheumatic diseases, Int J Rheum Dis', 182),\n",
       " ('International journal of systematic and evolutionary microbiology, Int. J. Syst. Evol. Microbiol.',\n",
       "  183),\n",
       " ('Iranian journal of basic medical sciences, Iran J Basic Med Sci', 184),\n",
       " ('Iranian journal of pathology, Iran J Pathol', 185),\n",
       " (\"Journal of Alzheimer's disease : JAD, J. Alzheimers Dis.\", 186),\n",
       " ('Journal of Cancer, J Cancer', 187),\n",
       " ('Journal of affective disorders, J Affect Disord', 188),\n",
       " ('Journal of agricultural and food chemistry, J. Agric. Food Chem.', 189),\n",
       " ('Journal of applied genetics, J. Appl. Genet.', 190),\n",
       " ('Journal of applied microbiology, J. Appl. Microbiol.', 191),\n",
       " ('Journal of bioinformatics and computational biology, J Bioinform Comput Biol',\n",
       "  192),\n",
       " ('Journal of biomedical informatics, J Biomed Inform', 193),\n",
       " ('Journal of biomedical semantics, J Biomed Semantics', 194),\n",
       " ('Journal of bioscience and bioengineering, J. Biosci. Bioeng.', 195),\n",
       " ('Journal of biotechnology, J. Biotechnol.', 196),\n",
       " ('Journal of cancer research and clinical oncology, J. Cancer Res. Clin. Oncol.',\n",
       "  197),\n",
       " ('Journal of cardiovascular translational research, J Cardiovasc Transl Res',\n",
       "  198),\n",
       " ('Journal of cellular and molecular medicine, J. Cell. Mol. Med.', 199),\n",
       " ('Journal of cellular biochemistry, J. Cell. Biochem.', 200),\n",
       " ('Journal of cellular physiology, J. Cell. Physiol.', 201),\n",
       " ('Journal of child psychology and psychiatry, and allied disciplines, J Child Psychol Psychiatry',\n",
       "  202),\n",
       " ('Journal of clinical neurology (Seoul, Korea), J Clin Neurol', 203),\n",
       " ('Journal of computational biology : a journal of computational molecular cell biology, J. Comput. Biol.',\n",
       "  204),\n",
       " ('Journal of economic entomology, J. Econ. Entomol.', 205),\n",
       " ('Journal of environmental science and health. Part. B, Pesticides, food contaminants, and agricultural wastes, J Environ Sci Health B',\n",
       "  206),\n",
       " ('Journal of experimental botany, J. Exp. Bot.', 207),\n",
       " ('Journal of fish biology, J. Fish Biol.', 208),\n",
       " ('Journal of genetics, J. Genet.', 209),\n",
       " ('Journal of global antimicrobial resistance, J Glob Antimicrob Resist', 210),\n",
       " ('Journal of hazardous materials, J. Hazard. Mater.', 211),\n",
       " ('Journal of immunological methods, J. Immunol. Methods', 212),\n",
       " ('Journal of immunology (Baltimore, Md. : 1950), J. Immunol.', 213),\n",
       " ('Journal of immunology research, J Immunol Res', 214),\n",
       " ('Journal of insect physiology, J. Insect Physiol.', 215),\n",
       " ('Journal of insect science (Online), J. Insect Sci.', 216),\n",
       " ('Journal of invertebrate pathology, J. Invertebr. Pathol.', 217),\n",
       " ('Journal of microbiological methods, J. Microbiol. Methods', 218),\n",
       " ('Journal of molecular evolution, J. Mol. Evol.', 219),\n",
       " ('Journal of molecular neuroscience : MN, J. Mol. Neurosci.', 220),\n",
       " ('Journal of neuroscience research, J. Neurosci. Res.', 221),\n",
       " ('Journal of orthopaedic surgery and research, J Orthop Surg Res', 222),\n",
       " ('Journal of plant research, J. Plant Res.', 223),\n",
       " ('Journal of psychopharmacology (Oxford, England), J. Psychopharmacol. (Oxford)',\n",
       "  224),\n",
       " ('Journal of the science of food and agriculture, J. Sci. Food Agric.', 225),\n",
       " ('Journal of theoretical biology, J. Theor. Biol.', 226),\n",
       " ('Journal of thoracic disease, J Thorac Dis', 227),\n",
       " ('Journal of translational medicine, J Transl Med', 228),\n",
       " ('Journal of virology, J. Virol.', 229),\n",
       " ('Lab on a chip, Lab Chip', 230),\n",
       " ('Le infezioni in medicina, Infez Med', 231),\n",
       " ('Letters in applied microbiology, Lett. Appl. Microbiol.', 232),\n",
       " ('Lipids, Lipids', 233),\n",
       " ('Mammalian genome : official journal of the International Mammalian Genome Society, Mamm. Genome',\n",
       "  234),\n",
       " ('Marine biotechnology (New York, N.Y.), Mar. Biotechnol.', 235),\n",
       " ('Mathematical biosciences, Math Biosci', 236),\n",
       " ('Medical science monitor : international medical journal of experimental and clinical research, Med. Sci. Monit.',\n",
       "  237),\n",
       " ('Medicinal chemistry (Shariqah (United Arab Emirates)), Med Chem', 238),\n",
       " ('Medicine, Medicine (Baltimore)', 239),\n",
       " ('Melanoma research, Melanoma Res.', 240),\n",
       " ('Methods (San Diego, Calif.), Methods', 241),\n",
       " ('Methods in molecular biology (Clifton, N.J.), Methods Mol. Biol.', 242),\n",
       " ('Microbial cell factories, Microb. Cell Fact.', 243),\n",
       " ('Microbial ecology, Microb. Ecol.', 244),\n",
       " ('Microbial pathogenesis, Microb. Pathog.', 245),\n",
       " ('Microbiological research, Microbiol. Res.', 246),\n",
       " ('Microbiology (Reading, England), Microbiology (Reading, Engl.)', 247),\n",
       " ('Microbiome, Microbiome', 248),\n",
       " ('Microorganisms, Microorganisms', 249),\n",
       " ('Molecular and cellular biochemistry, Mol. Cell. Biochem.', 250),\n",
       " ('Molecular and cellular neurosciences, Mol. Cell. Neurosci.', 251),\n",
       " ('Molecular and cellular probes, Mol. Cell. Probes', 252),\n",
       " ('Molecular bioSystems, Mol Biosyst', 253),\n",
       " ('Molecular biology and evolution, Mol. Biol. Evol.', 254),\n",
       " ('Molecular biology reports, Mol. Biol. Rep.', 255),\n",
       " ('Molecular ecology, Mol. Ecol.', 256),\n",
       " ('Molecular genetics & genomic medicine, Mol Genet Genomic Med', 257),\n",
       " ('Molecular genetics and genomics : MGG, Mol. Genet. Genomics', 258),\n",
       " ('Molecular human reproduction, Mol. Hum. Reprod.', 259),\n",
       " ('Molecular imaging and biology, Mol Imaging Biol', 260),\n",
       " ('Molecular medicine reports, Mol Med Rep', 261),\n",
       " ('Molecular microbiology, Mol. Microbiol.', 262),\n",
       " ('Molecular neurobiology, Mol. Neurobiol.', 263),\n",
       " ('Molecular oncology, Mol Oncol', 264),\n",
       " ('Molecular phylogenetics and evolution, Mol. Phylogenet. Evol.', 265),\n",
       " ('Molecular plant pathology, Mol. Plant Pathol.', 266),\n",
       " ('Molecules (Basel, Switzerland), Molecules', 267),\n",
       " ('Molekuliarnaia biologiia, Mol. Biol. (Mosk.)', 268),\n",
       " ('NPJ genomic medicine, NPJ Genom Med', 269),\n",
       " ('Nature genetics, Nat. Genet.', 270),\n",
       " ('Nephrology, dialysis, transplantation : official publication of the European Dialysis and Transplant Association - European Renal Association, Nephrol. Dial. Transplant.',\n",
       "  271),\n",
       " ('Neural development, Neural Dev', 272),\n",
       " ('Neurogenetics, Neurogenetics', 273),\n",
       " ('Neurological research, Neurol. Res.', 274),\n",
       " ('Neuropsychiatric disease and treatment, Neuropsychiatr Dis Treat', 275),\n",
       " ('Nitric oxide : biology and chemistry, Nitric Oxide', 276),\n",
       " ('Nucleic acids research, Nucleic Acids Res.', 277),\n",
       " ('Nutrients, Nutrients', 278),\n",
       " ('OncoTargets and therapy, Onco Targets Ther', 279),\n",
       " ('Oncogene, Oncogene', 280),\n",
       " ('Oncology letters, Oncol Lett', 281),\n",
       " ('Oncology reports, Oncol. Rep.', 282),\n",
       " ('Oncotarget, Oncotarget', 283),\n",
       " ('PLoS genetics, PLoS Genet.', 284),\n",
       " ('Pathologie-biologie, Pathol. Biol.', 285),\n",
       " ('Pathology oncology research : POR, Pathol. Oncol. Res.', 286),\n",
       " ('Pathology, research and practice, Pathol. Res. Pract.', 287),\n",
       " ('Pediatric research, Pediatr. Res.', 288),\n",
       " ('PeerJ, PeerJ', 289),\n",
       " ('Pest management science, Pest Manag. Sci.', 290),\n",
       " ('Pesticide biochemistry and physiology, Pestic Biochem Physiol', 291),\n",
       " ('Philosophical transactions of the Royal Society of London. Series B, Biological sciences, Philos. Trans. R. Soc. Lond., B, Biol. Sci.',\n",
       "  292),\n",
       " ('Physiological genomics, Physiol. Genomics', 293),\n",
       " ('Physiological research, Physiol Res', 294),\n",
       " ('Physiology and molecular biology of plants : an international journal of functional plant biology, Physiol Mol Biol Plants',\n",
       "  295),\n",
       " ('Phytopathology, Phytopathology', 296),\n",
       " ('Plant & cell physiology, Plant Cell Physiol.', 297),\n",
       " ('Plant biology (Stuttgart, Germany), Plant Biol (Stuttg)', 298),\n",
       " ('Plant biotechnology journal, Plant Biotechnol. J.', 299),\n",
       " ('Plant cell reports, Plant Cell Rep.', 300),\n",
       " ('Plant disease, Plant Dis.', 301),\n",
       " ('Plant methods, Plant Methods', 302),\n",
       " ('Plant molecular biology, Plant Mol. Biol.', 303),\n",
       " ('Plant physiology and biochemistry : PPB, Plant Physiol. Biochem.', 304),\n",
       " ('Plant physiology, Plant Physiol.', 305),\n",
       " ('Plant, cell & environment, Plant Cell Environ.', 306),\n",
       " ('Planta, Planta', 307),\n",
       " ('Plastic and reconstructive surgery, Plast. Reconstr. Surg.', 308),\n",
       " ('PloS one, PLoS ONE', 309),\n",
       " ('Polish journal of veterinary sciences, Pol J Vet Sci', 310),\n",
       " ('Poultry science, Poult. Sci.', 311),\n",
       " ('Proceedings of the National Academy of Sciences of the United States of America, Proc. Natl. Acad. Sci. U.S.A.',\n",
       "  312),\n",
       " ('Progress in neuro-psychopharmacology & biological psychiatry, Prog. Neuropsychopharmacol. Biol. Psychiatry',\n",
       "  313),\n",
       " ('Psychopharmacology, Psychopharmacology (Berl.)', 314),\n",
       " ('Radiation and environmental biophysics, Radiat Environ Biophys', 315),\n",
       " ('Radiation research, Radiat. Res.', 316),\n",
       " ('Research in microbiology, Res. Microbiol.', 317),\n",
       " ('Rice (New York, N.Y.), Rice (N Y)', 318),\n",
       " ('Scientific reports, Sci Rep', 319),\n",
       " ('Surgical oncology, Surg Oncol', 320),\n",
       " ('TAG. Theoretical and applied genetics. Theoretische und angewandte Genetik, Theor. Appl. Genet.',\n",
       "  321),\n",
       " ('Technology in cancer research & treatment, Technol. Cancer Res. Treat.',\n",
       "  322),\n",
       " ('The Journal of allergy and clinical immunology, J. Allergy Clin. Immunol.',\n",
       "  323),\n",
       " ('The Journal of biological chemistry, J. Biol. Chem.', 324),\n",
       " ('The Journal of general and applied microbiology, J. Gen. Appl. Microbiol.',\n",
       "  325),\n",
       " ('The Journal of international medical research, J. Int. Med. Res.', 326),\n",
       " ('The New phytologist, New Phytol.', 327),\n",
       " ('The Plant cell, Plant Cell', 328),\n",
       " ('The Plant journal : for cell and molecular biology, Plant J.', 329),\n",
       " ('The Science of the total environment, Sci. Total Environ.', 330),\n",
       " ('The journals of gerontology. Series A, Biological sciences and medical sciences, J. Gerontol. A Biol. Sci. Med. Sci.',\n",
       "  331),\n",
       " ('The world journal of biological psychiatry : the official journal of the World Federation of Societies of Biological Psychiatry, World J. Biol. Psychiatry',\n",
       "  332),\n",
       " ('Theriogenology, Theriogenology', 333),\n",
       " ('Toxicology and industrial health, Toxicol Ind Health', 334),\n",
       " ('Toxins, Toxins (Basel)', 335),\n",
       " ('Transcription, Transcription', 336),\n",
       " ('Translational psychiatry, Transl Psychiatry', 337),\n",
       " ('Transplantation proceedings, Transplant. Proc.', 338),\n",
       " ('Tree physiology, Tree Physiol.', 339),\n",
       " ('Tumori, Tumori', 340),\n",
       " ('Twin research and human genetics : the official journal of the International Society for Twin Studies, Twin Res Hum Genet',\n",
       "  341),\n",
       " ('Vascular, Vascular', 342),\n",
       " ('Veterinary immunology and immunopathology, Vet. Immunol. Immunopathol.',\n",
       "  343),\n",
       " ('Veterinary microbiology, Vet. Microbiol.', 344),\n",
       " ('Virology, Virology', 345),\n",
       " ('Virulence, Virulence', 346),\n",
       " ('Viruses, Viruses', 347),\n",
       " ('Water environment research : a research publication of the Water Environment Federation, Water Environ. Res.',\n",
       "  348),\n",
       " ('World journal of gastroenterology, World J. Gastroenterol.', 349),\n",
       " ('World journal of microbiology & biotechnology, World J. Microbiol. Biotechnol.',\n",
       "  350),\n",
       " ('World journal of surgical oncology, World J Surg Oncol', 351),\n",
       " ('Ying yong sheng tai xue bao = The journal of applied ecology, Ying Yong Sheng Tai Xue Bao',\n",
       "  352),\n",
       " ('Yonsei medical journal, Yonsei Med. J.', 353),\n",
       " ('Zeitschrift fur Naturforschung. C, Journal of biosciences, Z. Naturforsch., C, J. Biosci.',\n",
       "  354),\n",
       " ('Zoological science, Zool. Sci.', 355)]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn.execute(\"SELECT * FROM PUBLICATION\").fetchall()\n",
    "conn.execute(\"SELECT * FROM AUTEUR\").fetchall()\n",
    "conn.execute(\"SELECT * FROM JOURNAL_TABLE\").fetchall()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:42:01.180880Z",
     "start_time": "2020-06-24T13:42:01.159911Z"
    }
   },
   "outputs": [],
   "source": [
    "# Create Foreign Key\n",
    "# https://www.techonthenet.com/sqlite/foreign_keys/foreign_keys.php\n",
    "# https://www.sqlitetutorial.net/sqlite-foreign-key/\n",
    "# https://alvinalexander.com/android/sqlite-foreign-keys-example/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Faire des requêtes SQL\n",
    "\n",
    "- pour récupérer les abstracts des auteurs ayant au moins publiés deux fois.\n",
    "- pour récupérer un classement des journaux scientifiques par rapport aux nombres d'articles publiés\n",
    "- pour récupérer le nombre d'articles publiés par an"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T13:52:21.358672Z",
     "start_time": "2020-06-24T13:52:21.347184Z"
    }
   },
   "outputs": [],
   "source": [
    "# https://www.guru99.com/sqlite-query.html\n",
    "# https://www.sqlitetutorial.net/sqlite-count-function/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T14:17:05.154200Z",
     "start_time": "2020-06-24T14:17:04.936980Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(\"\\nAttention-Deficit Hyperactivity Disorder (ADHD) is a complex neurodevelopmental disorder (NDD) which may significantly impact on the affected individual's life. ADHD is acknowledged to have a high heritability component (70-80%). Recently, a meta-analysis of GWAS (Genome Wide Association Studies) has demonstrated the association of several independent loci. Our main aim here, is to apply PASCAL (pathway scoring algorithm), a new gene-based analysis (GBA) method, to the summary statistics obtained in this meta-analysis. PASCAL will take into account the linkage disequilibrium (LD) across genomic regions in a different way than the most commonly employed GBA methods (MAGMA or VEGAS (Versatile Gene-based Association Study)). In addition to PASCAL analysis a gene network and an enrichment analysis for KEGG and GO terms were carried out. Moreover, GENE2FUNC tool was employed to create gene expression heatmaps and to carry out a (DEG) (Differentially Expressed Gene) analysis using GTEX v7 and BrainSpan data.\\nPASCAL results have revealed the association of new loci with ADHD and it has also highlighted other genes previously reported by MAGMA analysis. PASCAL was able to discover new associations at a gene level for ADHD: FEZF1 (p-value: 2.2\\u2009×\\u200910-\\u20097) and FEZF1-AS1 (p-value: 4.58\\u2009×\\u200910-\\u20097). In addition, PASCAL has been able to highlight association of other genes that share the same LD block with some previously reported ADHD susceptibility genes. Gene network analysis has revealed several interactors with the associated ADHD genes and different GO and KEGG terms have been associated. In addition, GENE2FUNC has demonstrated the existence of several up and down regulated expression clusters when the associated genes and their interactors were considered.\\nPASCAL has been revealed as an efficient tool to extract additional information from previous GWAS using their summary statistics. This study has identified novel ADHD associated genes that were not previously reported when other GBA methods were employed. Moreover, a biological insight into the biological function of the ADHD associated genes across brain regions and neurodevelopmental stages is provided.\\n\",\n",
       "  2),\n",
       " ('\\nGene expression data were analysed using bioinformatic tools to demonstrate molecular mechanisms underlying the glioma CpG island methylator phenotype (CIMP). A gene expression data set (accession no.\\xa0GSE30336) was downloaded from Gene Expression Omnibus, including 36\\xa0CIMP+ and 16\\xa0CIMP-\\xa0glioma samples. Differential analysis was performed for CIMP+ vs.\\xa0CIMP‑ samples using the limma package in R. Functional enrichment analysis was subsequently conducted for differentially expressed genes (DEGs) using Database for Annotation, Visualization and Integration Discovery. Protein‑protein interaction (PPI) networks were constructed for upregulated and downregulated genes with information from STRING. MicroRNAs (miRNAs) targeting DEGs were also predicted using WebGestalt. A total of 439\\xa0DEGs were identified, including 214\\xa0upregulated and 198\\xa0downregulated genes. The upregulated genes were involved in extracellular matrix organisation, defence and immune response, collagen fibril organisation and regulation of cell motion and the downregulated genes in cell adhesion, sensory organ development, regulation of system process, neuron differentiation and membrane organisation. A PPI network containing 134\\xa0nodes and 314\\xa0edges was constructed from the upregulated genes, whereas a PPI network consisting of 85\\xa0nodes and 80\\xa0edges was obtained from the downregulated genes. miRNAs regulating upregulated and downregulated genes were predicted, including miRNA‑124a and miRNA‑34a. Numerous key genes associated with glioma CIMP were identified in the present study. These findings may advance the understanding of glioma and facilitate the development of appropriate therapies.\\n',\n",
       "  2),\n",
       " ('\\nDetermining which reference genes have the highest stability, and are therefore appropriate for normalising data, is a crucial step in the design of real-time quantitative PCR (qPCR) gene expression studies. This is particularly warranted in non-model and ecologically important species for which appropriate reference genes are lacking, such as the mallard--a key reservoir of many diseases with relevance for human and livestock health. Previous studies assessing gene expression changes as a consequence of infection in mallards have nearly universally used β-actin and/or GAPDH as reference genes without confirming their suitability as normalisers. The use of reference genes at random, without regard for stability of expression across treatment groups, can result in erroneous interpretation of data. Here, eleven putative reference genes for use in gene expression studies of the mallard were evaluated, across six different tissues, using a low pathogenic avian influenza A virus infection model. Tissue type influenced the selection of reference genes, whereby different genes were stable in blood, spleen, lung, gastrointestinal tract and colon. β-actin and GAPDH generally displayed low stability and are therefore inappropriate reference genes in many cases. The use of different algorithms (GeNorm and NormFinder) affected stability rankings, but for both algorithms it was possible to find a combination of two stable reference genes with which to normalise qPCR data in mallards. These results highlight the importance of validating the choice of normalising reference genes before conducting gene expression studies in ducks. The fact that nearly all previous studies of the influence of pathogen infection on mallard gene expression have used a single, non-validated reference gene is problematic. The toolkit of putative reference genes provided here offers a solid foundation for future studies of gene expression in mallards and other waterfowl.\\n',\n",
       "  2),\n",
       " ('\\nTriple-negative breast cancer (TNBC) is a special subtype of breast cancer (BC) with poor prognosis. Although some molecular mechanisms of TNBC have been elucidated, the efficacy of current treatments is limited. Therefore, it is urgently demanded to screen for novel biomarkers and drug targets for TNBC. In this study, we obtained four independent data sets (GSE76250, GSE31448, GSE43358, and METABRIC) from the Gene Expression Omnibus (GEO) database and the cBioPortal website. In the GSE76250 data set, 890 differentially expressed genes were identified and weighted gene co-expression network analysis was performed based on them. Then, two preserved modules associated with the KI67 score were detected. Gene ontology and pathway enrichment analyses showed genes in the modules participated in some cancer-related biological processes or pathways. Non-SMC condensin I complex subunit G (NCAPG) and ATP-binding cassette subfamily A member 9 (ABCA9) were identified as hub genes of the modules, and the significance of hub genes was validated in the GSE43358 data set. Finally, their prognostic value was assessed by survival analysis. These findings suggested that NCAPG and ABCA9 may be the key genes of TNBC. Moreover, ABCA9 was first reported in TNBC. They deserved further studies.\\n© 2019 Wiley Periodicals, Inc.\\n',\n",
       "  2),\n",
       " ('\\nγδ T cells recognize a wide variety of ligands in mammals, among them members of the butyrophilin (BTN) family. Nothing is known about γδ T cell ligands in chickens, despite there being many such cells in blood and lymphoid tissues, as well as in mucosal surfaces. The major histocompatibility complex (MHC) of chickens was discovered because of polymorphic BG genes, part of the BTN family. All but two BG genes are located in the BG region, oriented head-to-tail so that unequal crossing-over has led to copy number variation (CNV) as well as hybrid (chimeric) genes, making it difficult to identify true alleles. One approach is to examine BG genes expressed in particular cell types, which likely have the same functions in different BG haplotypes and thus can be considered \"functional alleles.\" We cloned nearly full-length BG transcripts from peripheral T cells of four haplotypes (B2, B15, B19, and B21), and compared them to the BG genes of the B12 haplotype that previously were studied in detail. A dominant BG gene was found in each haplotype, but with significant levels of subdominant transcripts in three haplotypes (B2, B15, and B19). For three haplotypes (B15, B19, and B21), most sequences are closely-related to BG8, BG9, and BG12 from the B12 haplotype. We found that variation in the extracellular immunoglobulin-variable-like (Ig-V) domain is mostly localized to the membrane distal loops but without evidence for selection. However, variation in the cytoplasmic tail composed of many amino acid heptad repeats does appear to be selected (although not obviously localized), consistent with an intriguing clustering of charged and polar residues in an apparent α-helical coiled-coil. By contrast, the dominantly-expressed BG gene in the B2 haplotype is identical to BG13 from the B12 haplotype, and most of the subdominant sequences are from the BG5-BG7-BG11 clade. Moreover, alternative splicing leading to intron read-through results in dramatically truncated cytoplasmic tails, particularly for the dominantly-expressed BG gene of the B2 haplotype. The approach of examining \"functional alleles\" has yielded interesting data for closely-related genes, but also thrown up unexpected findings for at least one haplotype.\\n',\n",
       "  2),\n",
       " ('\\nThe Salt Overly Sensitive (SOS) pathway in Arabidopsis thaliana plays important roles in maintaining appropriate ion homeostasis in the cytoplasm and regulating plant tolerance to salinity. However, little is known about the details regarding SOS family genes in the tuber mustard crop (Brassica juncea var. tumida). Here, 12 BjSOS family genes were identified in the B. juncea var. tumida genome including two homologous genes of SOS1, one and three homologs of SOS2 and SOS3, two homologs of SOS4, two homologs of SOS5 and two homologs of SOS6, respectively. The results of conserved motif analysis showed that these SOS homologs contained similar protein structures. By analyzing the cis-elements in the promoters of those BjSOS genes, several hormone- and stress-related cis-elements were found. The results of gene expression analysis showed that the homologous genes were induced by abiotic stress and pathogen. These findings indicate that BjSOS genes play crucial roles in the plant response to biotic and abiotic stresses. This study provides valuable information for further investigations of BjSOS genes in tuber mustard.\\n',\n",
       "  2),\n",
       " (\"\\nGenes encoding casein proteins are important candidates for milk composition traits in mammals. In the case of the domestic horse, our knowledge of casein genes is limited mainly to coding sequence variants. This study involved screening for polymorphism in 5'-flanking regions of four genes encoding equine caseins (CSN1S1, CSN1S2, CSN2, and CSN3) and making a preliminary assessment of their effect on the gene expression (on the mRNA and protein levels) and milk composition traits in selected horse breeds. Altogether, 23 polymorphisms (21 described previously SNPs and two novel InDels) were found in the studied sequences, the majority of which are common in various horse breeds. Statistical analysis revealed that some are putatively associated with gene expression or milk composition\\xa0-\\xa0for example, the c.-2047_-2048insAT polymorphism (CSN1S1) turns out to be related to the total milk protein content in Polish Primitive Horse\\xa0(p\\u2009<\\u20090.05), whereas c.-2105C>G SNP (CSN2) is related to beta-casein relative mRNA level and milk lactose concentration in the Polish Coldblood Horse breed (p\\u2009<\\u20090.05). We have also found significant effects of horse breed and lactation time-point on gene expression and mare's milk composition. Our study indicates that the 5'-regulatory regions of genes encoding casein proteins are interesting targets for functional studies of their expression and the composition traits of mare's milk.\\n\",\n",
       "  2),\n",
       " ('\\nThis study was aimed at identifying differentially expressed genes (DEGs) with copy number changes in gastric cancer (GC) pathogenesis. Microarray data GSE33429, including array-based comparative genomic hybridization and gene expression profiles, were obtained. DEGs were screened between GC and adjacent noncancerous tissues. Genes located at Minimum Common Regions (MCRs) were identified, and overlapped genes between DEGs and genes with amplification or deletion were identified. Gene Ontology function and pathway enrichment analysis of DEGs were performed. A protein-protein interaction network for DEGs was built, and significant modules were mined from the network. Functional annotation of genes in modules was also performed. A total of 677 up- and 583 downregulated DEGs were identified, including 37 overexpressed genes located at gained MCRs and 28 downregulated genes located at deleted MCRs. In significant modules, upregulated genes with amplification, including DSN1 (MIS12 kinetochore complex component), MAPRE1 (microtubule-associated protein, RP/EB family, member 1), TPX2 (microtubule-associated), UBE2C (ubiquitin-conjugating enzyme E2C), and MYBL2 (v-myb avian myeloblastosis viral oncogene homolog-like 2), were associated with cell cycle, but downregulated genes with deletion, including UGT2B15 (UDP glucuronosyltransferase 2 family, polypeptide B15), UGT2B17 (UDP glucuronosyltransferase 2 family, polypeptide B17), ADH1B (alcohol dehydrogenase 1B), and ADH1A (alcohol dehydrogenase 1A), were related to metabolism. The identified genes DSN1, MAPRE1, TPX2, UBE2C, and MYBL2 located at gained MCRs and UGT2B15, UGT2B17, ADH1B, and ADH1A located at deleted MCRs may play an important role in GC progression through regulating cell cycle and metabolism.\\n',\n",
       "  2),\n",
       " ('\\nThe aim of the present study was to analyze lung adenocarcinoma-associated microarray data and identify potentially crucial genes. The gene expression profiles were downloaded from the Gene Expression Omnibus database and 6 datasets, of which 2 were discarded and 4 were retained, were preprocessed using packages in the R computing language. Subsequently, Gene Set Enrichment Analysis (GSEA) and meta-analysis was used to screen the common pathways and differentially expressed genes at the transcriptional level. The genes detected from GSEA through The Cancer Genome Atlas databases were subsequently examined, and the crucial genes by survival data were identified. Pathways of the crucial genes were obtained using the Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway of the online website Database for Annotation, Visualization and Integrated Discovery (DAVID) tool, and the pathways of crucial genes that were upregulated or downregulated were matched using the Venn method to identify the common crucial pathways. Furthermore, on the basis of the common crucial pathways, key genes that are closely associated with the development and progression of lung adenocarcinoma were identified with the KEGG pathway of DAVID. Additional information was obtained through Gene Ontology annotation. A total of two key pathways, including cell cycle and DNA replication, as well as 12 key genes [DNA polymerase δ subunit 2, DNA replication licensing factor MCM4, MCM6, mitotic checkpoint serine/threonine-protein kinase BUB1, BUB1β, mitotic spindle assembly checkpoint protein MAD2A, dual specificity protein kinase TTK, M-phase inducer phosphatase 1, cell division control protein 45 homolog, cyclin-dependent kinase inhibitor 1C, pituitary tumor-transforming gene 1 protein and polo-like kinase 1] were identified. These key pathways and genes may be studied in future studies involving gene transfection/knockdown, which may provide insights into the prognosis of lung adenocarcinoma. Additional studies are required to confirm their biological function.\\n',\n",
       "  3),\n",
       " (\"\\nAeromonas sp. are opportunistic pathogenic bacteria which are associated with various diseases in ornamental fish, aquaculture raised species and wild fisheries. In our study, antimicrobial resistance patterns, antimicrobial resistance genes and class 1 integron gene cassettes of 52 guppy-borne Aeromonas sp. were examined. The isolates were identified as A. veronii (n\\xa0=\\xa034), A. dhakensis (n\\xa0=\\xa010), A. hydrophila (n\\xa0=\\xa03), A. caviae (n\\xa0=\\xa03) and A. enteropelogenes (n\\xa0=\\xa02) by gyrB gene sequencing. Every isolate was resistant to at least four antimicrobials in disc diffusion test. The resistance to amoxicillin, nalidixic acid and oxytetracycline was 100% among the tested isolates. 92·30, 76·92, 71·15, 51·92, 51·92 and 50·00% of the isolates were resistant to ampicillin, rifampicin, imipenem, cephalothin, tetracycline and trimethoprim respectively. The multiple antibiotic resistance index values ranged from 0·28 to 0·67. PCR amplification of antimicrobial resistance genes implied the occurrence of tetracycline resistance (tetA (65·39%), tetE (25·00%) and tetB (15·38%)), plasmid-mediated quinolone resistance (qnrS (26·92%) and qnrB (17·31%)) and aminoglycoside resistance (aphaAI-IAB (7·69%) and aac (6')-Ib (3·84%)) genes in the isolates. The IntI gene was positive for 36·54% of the isolates and four class 1 integron gene cassette profiles (aadA2, qacE2-orfD, aadA2-catB2 and dfrA12-aadA2) were identified. These data suggest that ornamental guppy can be a reservoir of multidrug-resistant Aeromonas sp. which comprise different antimicrobial resistance genes and class 1 integrons. SIGNIFICANCE AND IMPACT OF THE STUDY: Antimicrobial resistance genes and integron gene cassettes of ornamental fish-borne aeromonads are poorly studied. The antimicrobial resistance patterns, antimicrobial resistance genes and class 1 integron gene cassettes of Aeromonas sp. isolated from ornamental guppy were characterized for the first time in Korea. The incidence of different antimicrobial resistance genes and class 1 integron gene cassettes were observed in multidrug-resistant Aeromonas isolates. This result suggests that better management practices are necessary to prevent and address the serious consequences of indiscriminate and inappropriate antimicrobial use, and the distribution of multidrug-resistant bacteria.\\n© 2019 The Society for Applied Microbiology.\\n\",\n",
       "  2),\n",
       " ('\\nRapid adaptation/accommodation to changing environments largely contributes to maximal survival of invaders during biological invasions, usually leading to success in crossing multiple barriers and finally in varied environments in recipient habitats. Gene expression is one of the most important and rapid ways during responses to environmental stresses. Selection of proper reference genes is the crucial prerequisite for gene expression analysis using the common approach, real-time quantitative PCR (RT-qPCR). Here we identified eight candidate novel reference genes from the RNA-Seq data in an invasive model ascidian Ciona savignyi under temperature and salinity stresses. Subsequently, the expression stability of these eight novel reference genes, as well as other six traditionally used reference genes, was evaluated using RT-qPCR and comprehensive tool RefFinder. Under the temperature stress, two traditional reference genes, ribosomal proteins S15 and L17 (RPS15, RPL17), and one novel gene Ras homolog A (RhoA), were recommended as the top three stable genes, which can be used to normalize target genes with a high and moderate expression level, respectively. Under the salinity stress, transmembrane 9 superfamily member (TMN), MOB kinase activator 1A-like gene (MOB) and ubiquitin-conjugating enzyme (UBQ2) were suggested as the top three stable genes. On the other hand, several commonly used reference genes such as α-tubulin (TubA), β-tubulin (TubB) and glyceraldehyde-3-phosphate dehydrogenase (GAPDH) showed unstable expressions, thus these genes should not be used as internal controls for gene expression analysis. We also tested the expression level of an important stress response gene, large proline-rich protein bag6-like gene (BAG) using different reference genes. As expected, we observed different results and conclusions when using different normalization methods, thus suggesting the importance of selection of proper reference genes and associated normalization methods. Our results provide a valuable reference gene resource for the normalization of gene expression in the study of environmental adaptation/accommodation during biological invasions using C. savignyi as a model.\\n',\n",
       "  2),\n",
       " ('\\nThis is the first study which describes a unique procedure of isolating of high-quality, intact RNA from strawberry leaves of Xanthomonas fragariae, three most suitable reference genes, crucial for the normalization of RT-qPCR data for this pathogen and accurate expression analysis of target genes. In our study, various mathematic algorithms: NormFinder geNorm, BestKeeper, the delta CT method, RefFinder were adopted for validation of most stable reference genes from nine candidate genes (ffh, glyA, gyrA, gyrB, proC, pykA, recA, rpoB, rpoD). The analyses allowing to select three most suitable pioneer reference genes, gyrB, ffh, and pykA, that we recommend for the normalization of RT-qPCR data and for the study of the expression of target genes in Xf. Moreover, their combination as references allowed for an accurate expression analysis and computation of the fold change of the flhF and iroN2 genes in Xf. These two genes are important for the success of the colonization of plant tissue and pathogenicity and sequences of primers designed to study these genes, are presented.\\n',\n",
       "  2),\n",
       " ('\\nMelon (Cucumis melo L.) is an attractive model plant for investigating fruit development because of its morphological, physiological, and biochemical diversity. Quantification of gene expression by quantitative reverse transcription polymerase chain reaction (qRT-PCR) with stably expressed reference genes for normalization can effectively elucidate the biological functions of genes that regulate fruit development. However, the reference genes for data normalization in melon fruits have not yet been systematically validated. This study aims to assess the suitability of 20 genes for their potential use as reference genes in melon fruits. Expression variations of these genes were measured in 24 samples that represented different developmental stages of fertilized and parthenocarpic melon fruits by qRT-PCR analysis. GeNorm identified ribosomal protein L (CmRPL) and cytosolic ribosomal protein S15 (CmRPS15) as the best pair of reference genes, and as many as five genes including CmRPL, CmRPS15, TIP41-like family protein (CmTIP41), cyclophilin ROC7 (CmCYP7), and ADP ribosylation factor 1 (CmADP) were required for more reliable normalization. NormFinder ranked CmRPS15 as the best single reference gene, and RAN GTPase gene family (CmRAN) and TATA-box binding protein (CmTBP2) as the best combination of reference genes in melon fruits. Their effectiveness was further validated by parallel analyses on the activities of soluble acid invertase and sucrose phosphate synthase, and expression profiles of their respective encoding genes CmAIN2 and CmSPS1, as well as sucrose contents during melon fruit ripening. The validated reference genes will help to improve the accuracy of gene expression studies in melon fruits. \\n',\n",
       "  3),\n",
       " ('\\nPapillary thyroid carcinoma (PTC), which is the most common endocrine malignancy, has been steadily increasing worldwide in incidence over the years, while mechanisms underlying the pathogenesis and diagnostic for PTC are incomplete. The purpose of this study is to identify potential biomarkers for diagnosis of PTC, and provide new insights into pathogenesis of PTC.\\nBased on weighted gene co-expression network analysis, Robust Rank Aggregation, functional annotation, GSEA and DNA methylation, were employed for investigating potential biomarkers for diagnosis of PTC.\\nBlack and turquoise modules were identified in the gene co-expression network constructed by 1807 DEGs that from 6 eligible gene expression profiles of Gene Expression Omnibus database based on Robust Rank Aggregation and weighted gene co-expression network analysis. Hub genes were significantly down-regulated and the expression levels of the hub genes were different in different stages in hub gene verification. ROC curves indicated all hub genes had good diagnostic value for PTC (except for ABCA6 AUC\\u2009=\\u200989.5%, the 15 genes with AUC\\u2009>\\u200990%). Methylation analysis showed that hub gene verification ABCA6, ACACB, RMDN1 and TFPI were identified as differentially methylated genes, and the decreased expression level of these genes may relate to abnormal DNA methylation. Moreover, the expression levels of 8 top hub genes were correlated with tumor purity and tumor-infiltrating immune cells. These findings, including functional annotations and GSEA provide new insights into pathogenesis of PTC.\\nThe hub genes and methylation of hub genes may as potential biomarkers provide new insights for diagnosis of PTC, and all these findings may be the direction to study the mechanisms underlying of PTC in the future.\\n',\n",
       "  2),\n",
       " ('\\nPancreatic cancer is one of the malignant tumors that threaten human health.\\nThe gene expression profiles of GSE15471, GSE19650, GSE32676 and GSE71989 were downloaded from the gene expression omnibus database including pancreatic cancer and normal samples. The differentially expressed genes between the two types of samples were identified with the Limma package using R language. The gene ontology functional and pathway enrichment analyses of differentially-expressed genes were performed by the DAVID software followed by the construction of a protein-protein interaction network. Hub gene identification was performed by the plug-in cytoHubba in cytoscape software, and the reliability and survival analysis of hub genes was carried out in The Cancer Genome Atlas gene expression data.\\nThe 138 differentially expressed genes were significantly enriched in biological processes including cell migration, cell adhesion and several pathways, mainly associated with extracellular matrix-receptor interaction and focal adhesion pathway in pancreatic cancer. The top hub genes, namely thrombospondin 1, DNA topoisomerase II alpha, syndecan 1, maternal embryonic leucine zipper kinase and proto-oncogene receptor tyrosine kinase Met were identified from the protein-protein interaction network. The expression levels of hub genes were consistent with data obtained in The Cancer Genome Atlas. DNA topoisomerase II alpha, syndecan 1, maternal embryonic leucine zipper kinase and proto-oncogene receptor tyrosine kinase Met were significantly linked with poor survival in pancreatic adenocarcinoma.\\nThese hub genes may be used as potential targets for pancreatic cancer diagnosis and treatment.\\n',\n",
       "  3),\n",
       " (\"\\nGene-expression data is being widely used for various clinical research. It represents expression levels of thousands of genes across the various experimental conditions simultaneously. Mining conditions specific hub genes from gene-expression data is a challenging task. Conditions specific hub genes signify the functional behaviour of bicluster across the subset of conditions and can act as prognostic or diagnostic markers of the diseases. In this study, the authors have introduced a new approach for identifying conditions specific hub genes from the RNA-Seq data using a biclustering algorithm. In the proposed approach, efficient 'runibic' biclustering algorithm, the concept of gene co-expression network and concept of protein-protein interaction network have been used for getting better performance. The result shows that the proposed approach extracts biologically significant conditions specific hub genes which play an important role in various biological processes and pathways. These conditions specific hub genes can be used as prognostic or diagnostic biomarkers. Conditions specific hub genes will be helpful to reduce the analysis time and increase the accuracy of further research. Also, they summarised application of the proposed approach to the drug discovery process.\\n\",\n",
       "  2),\n",
       " ('\\nQuantitative Real-Time PCR (qPCR) is a preferred and reliable method for accurate quantification of gene expression to understand precise gene functions. A total of 25 candidate reference genes including traditional and new generation reference genes were selected and evaluated in a diverse set of chickpea samples. The samples used in this study included nine chickpea genotypes (Cicer spp.) comprising of cultivated and wild species, six abiotic stress treatments (drought, salinity, high vapor pressure deficit, abscisic acid, cold and heat shock), and five diverse tissues (leaf, root, flower, seedlings and seed). The geNorm, NormFinder and RefFinder algorithms used to identify stably expressed genes in four sample sets revealed stable expression of UCP and G6PD genes across genotypes, while TIP41 and CAC were highly stable under abiotic stress conditions. While PP2A and ABCT genes were ranked as best for different tissues, ABCT, UCP and CAC were most stable across all samples. This study demonstrated the usefulness of new generation reference genes for more accurate qPCR based gene expression quantification in cultivated as well as wild chickpea species. Validation of the best reference genes was carried out by studying their impact on normalization of aquaporin genes PIP1;4 and TIP3;1, in three contrasting chickpea genotypes under high vapor pressure deficit (VPD) treatment. The chickpea TIP3;1 gene got significantly up regulated under high VPD conditions with higher relative expression in the drought susceptible genotype, confirming the suitability of the selected reference genes for expression analysis. This is the first comprehensive study on the stability of the new generation reference genes for qPCR studies in chickpea across species, different tissues and abiotic stresses. \\n',\n",
       "  2),\n",
       " ('\\nCurrently, more than 150 million people worldwide suffer from lymphedema. It is a chronic progressive disease characterized by high-protein edema of various parts of the body due to defects in lymphatic drainage. Molecular-genetic mechanisms of the disease are still poorly understood. Beginning of a clinical manifestation of primary lymphedema in middle age and the development of secondary lymphedema after treatment of breast cancer can be genetically determined. Disruption of endothelial cell apoptosis can be considered as one of the factors contributing to the development of lymphedema. However, a study of the relationship between genes associated with lymphedema and genes involved in endothelial apoptosis, in the associative gene network was not previously conducted.\\nIn the current work, we used well-known methods (ToppGene and Endeavour), as well as methods previously developed by us, to prioritize genes involved in endothelial apoptosis and to find potential participants of molecular-genetic mechanisms of lymphedema among them. Original methods of prioritization took into account the overrepresented Gene Ontology biological processes, the centrality of vertices in the associative gene network, describing the interactions of endothelial apoptosis genes with genes associated with lymphedema, and the association of the analyzed genes with diseases that are comorbid to lymphedema.\\nAn assessment of the quality of prioritization was performed using criteria, which involved an analysis of the enrichment of the top-most priority genes by genes, which are known to have simultaneous interactions with lymphedema and endothelial cell apoptosis, as well as by genes differentially expressed in murine model of lymphedema. In particular, among genes involved in endothelial apoptosis, KDR, TNF, TEK, BMPR2, SERPINE1, IL10, CD40LG, CCL2, FASLG and ABL1 had the highest priority. The identified priority genes can be considered as candidates for genotyping in the studies involving the search for associations with lymphedema.\\nAnalysis of interactions of these genes in the associative gene network of lymphedema can improve understanding of mechanisms of interaction between endothelial apoptosis and lymphangiogenesis, and shed light on the role of disturbance of these processes in the development of edema, chronic inflammation and connective tissue transformation during the progression of the disease.\\n',\n",
       "  2),\n",
       " ('\\nOxidative stress is the process by which reactive molecules and free radicals are formed in cells. In this study, we report the blood-based gene expression profile of oxidative stress and antioxidant genes for identifying surrogate markers of liver tissue in\\xa0chronic hepatitis C (CHC) patients by using real-time PCR. A total of 144 untreated patients diagnosed with CHC having genotype 3a and 20 healthy controls were selected for the present study. Liver biopsy staging and grading of CHC patients were performed using the METAVIR score. Total RNA was extracted from liver tissue and blood samples, followed by cDNA synthesis and real-time PCR. The relative expression of genes was calculated using the ΔΔCt method. The expression profile of 84 genes associated with oxidative stress and antioxidants was determined in liver tissue and blood samples. In liver tissue, 46 differentially expressed genes (upregulated, 27; downregulated, 19) were identified in CHC patients compared to normal samples. In blood, 61 genes (upregulated, 51; downregulated; 10) were significantly expressed in CHC patients. A comparison of gene expression in liver and whole blood showed that 20 genes were expressed in a similar manner in the liver and blood. The expression levels of commonly expressed liver and blood-based genes were also correlated with clinical factors in CHC patients. A receiver operating curve (ROC) analysis of oxidative stress genes (ALB, CAT, DHCR24, GPX7, PRDX5, and MBL2) showed that infections in patients with CHC can be distinguished from healthy controls. In conclusion, blood-based gene expression can reflect the behavior of oxidative stress genes in liver tissue, and this blood-based gene expression study in CHC patients explores new blood-based non-invasive biomarkers that represent liver damage.\\n',\n",
       "  2),\n",
       " ('\\nOsteosarcoma is a common and highly malignant tumour in children and teenagers that is characterized by drug resistance and high metastatic potential. Patients often develop pulmonary metastasis and have a low survival rate. However, the mechanistic basis for pulmonary metastasis remains unclear. To identify key gene and pathways associated with pulmonary metastasis of osteosarcoma, the authors downloaded the gene expression dataset GSE85537 and obtained the differentially expressed genes (DEGs) by analyzing high‑throughput gene expression in primary tumours and lung metastases. Subsequently, the authors performed gene ontology (GO) and Kyoto Encyclopedia of Genes and Genomes pathway (KEGG) enrichment analyses and a protein‑protein interaction (PPI) network was constructed and analyzed by Cytoscape software. In total, 2,493 genes were identified as DEGs. Of these, 485 genes (19.45%) were upregulated, and the remaining 2,008 genes (80.55%) were downregulated. The authors identified the predominant GO categories and KEGG pathways that were significantly over‑represented in the metastatic OS samples compared with the non‑metastatic OS samples. A PPI network was constructed, and the results indicated that ALB, EGFR, INS, IL6, CDH1, FYN, ERBB2, IL8, CXCL12 and RAC2 were the top 10 core genes. The enrichment analyses of the genes involved in the top three signiﬁcant modules demonstrated that the DEGs were principally related to neuroactive ligand‑receptor interaction, the Rap1 signaling pathway, and protein digestion and absorption. Together, these data elucidated the molecular mechanisms of OS patients with pulmonary metastasis and provide potential therapeutic targets. However, further experimental studies are needed to confirm these results.\\n',\n",
       "  2),\n",
       " ('\\nOld and young duplicate genes have been reported in some organisms. However, little is known about the properties of old and young duplicate genes in Arachis. Here, we have identified old and young duplicate genes in Arachis duranensis, and analyzed the evolution, gene complexity, gene expression pattern, and functional divergence between old and young duplicate genes. Our results showed different evolutionary, gene complexity and gene expression patterns, as well as differing correlations between old and young duplicate genes. Gene ontology results showed that old duplicate genes play a crucial role in lipid and amino acid biosynthesis and the oxidation-reduction process and that young duplicate genes are preferentially involved in photosynthesis and response to biotic stimulus. Transcriptome data sets revealed that most old and young duplicate genes had asymmetric function, and only a few duplicate genes exhibited symmetric function under drought and nematode stress. We found that old duplicate genes are preferentially involved in lipid and amino acid metabolism and response to abiotic stress, while young duplicate genes are likely to participate in photosynthesis and response to biotic stress. This work provides a better understanding of the evolution and functional divergence of old and young duplicate genes in A. duranensis.\\n',\n",
       "  2),\n",
       " ('\\nKeshan disease (KD) is an endemic cardiomyopathy with high mortality. Selenium (Se) deficiency is closely related to KD, while magnesium (Mg) plays many critical roles in the cardiovascular function. The molecular mechanism of KD pathogenesis is still unclear. Until now, we have not found any studies investigating the association between Se- or Mg-related genes and KD. In this study, oligonucleotide microarray analysis was used to identify the differentially expressed genes in the peripheral blood mononuclear cells between KD patients and normal controls. Next, human metabolome database (HMDB) was used to screen Se- and Mg-related genes. Function classification, gene pathway, and interaction network of Se- and Mg-related genes in KD peripheral blood mononuclear cells were defined by FunRich (functional enrichment analysis tool). Among 83 differentially expressed genes, five Se-related (DIO2, GPX1, GPX2, GPX4, and GPX7) and five Mg-related (ACSL6, EYA4, IDH2, PPM1A, and STK11) genes were recognized from HMDB. Two significant biological processes (energy pathways and metabolism), one molecular function (peroxidase activity), one biological pathway (glutathione redox reactions I), and one gene interaction network were constituted from Se-related and Mg-related genes. Se-related gene DIO2 and Mg-related genes STK11 and IDH2 may have key roles in the myocardial dysfunction of KD. However, we still have not obtained any interaction between Se-related gene and Mg-related gene. The interactions between RPS6KB1, PTEN, ATM, HSP90AA1, SNRK, PRKAA2, SMARCA4, HSPA1A, and STK11 may play important roles in the abnormal cardiac function of KD.\\n',\n",
       "  2),\n",
       " ('\\nThis study aim to identify the core pathogenic genes and explore the potential molecular mechanisms of human coronary artery disease (CAD).\\nTwo gene profiles of epicardial adipose tissue from CAD patients including GSE 18612 and GSE 64554 were downloaded and integrated by R software packages. All the coexpression of deferentially expressed genes (DEGs) were picked out and analyzed by DAVID online bioinformatic tools. In addition, the DEGs were totally typed into protein-protein interaction (PPI) networks to get the interaction data among all coexpression genes. Pictures were drawn by cytoscape software with the PPI networks data. CytoHubba were used to predict the hub genes by degree analysis. Finally all the top 10 hub genes and prediction genes in Molecular complex detection were analyzed by Gene ontology and Kyoto encyclopedia of genes and genomes pathway analysis. qRT-PCR were used to identified all the 10 hub genes.\\nThe top 10 hub genes calculated by the degree method were AKT1, MYC, EGFR, ACTB, CDC42, IGF1, FGF2, CXCR4, MMP2 and LYN, which relevant with the focal adhesion pathway. Module analysis revealed that the focal adhesion was also acted an important role in CAD, which was consistence with cytoHubba. All the top 10 hub genes were verified by qRT-PCR which presented that AKT1, EGFR, CDC42, FGF2, and MMP2 were significantly decreased in epicardial adipose tissue of CAD samples (p < 0.05) and MYC, ACTB, IGF1, CXCR4, and LYN were significantly increased (p < 0.05).\\nThese candidate genes could be used as potential diagnostic biomarkers and therapeutic targets of CAD.\\nCopyright © 2019 Weitie Wang et al.\\n',\n",
       "  2),\n",
       " ('\\nWe aimed to explore potential molecular mechanisms of clear cell renal cell carcinoma (ccRCC) and provide candidate target genes for ccRCC gene therapy.\\nThis is a bioinformatics-based study. Microarray datasets of GSE6344, GSE781 and GSE53000 were downloaded from Gene Expression Omnibus database. Using meta-analysis, differentially expressed genes (DEGs) were identified between ccRCC and normal samples, followed by Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway and Gene Ontology (GO) function analyses. Then, protein-protein interaction (PPI) networks and modules were investigated. Furthermore, miRNAs-target gene regulatory network was constructed.\\nTotal of 511 up-regulated and 444 down-regulated DEGs were determined in the present gene expression microarray data meta-analysis. These DEGs were enriched in functions like immune system process and pathways like Toll-like receptor signaling pathway. PPI network and eight modules were further constructed. A total of 10 outstanding DEGs including TYRO protein tyrosine kinase binding protein (TYROBP), interferon regulatory factor 7 (IRF7) and PPARG co-activator 1 alpha (PPARGC1A) were detected in PPI network. Furthermore, the miRNAs-target gene regulation analyses showed that miR-412 and miR-199b respectively targeted IRF7 and PPARGC1A to regulate the immune response in ccRCC.\\nTYROBP, IRF7 and PPARGC1A might play important roles in ccRCC via taking part in the immune system process.\\nCopyright© by Royan Institute. All rights reserved.\\n',\n",
       "  2),\n",
       " ('\\nDiaphania caesalis (Walker) is an important boring insect mainly distributed in subtropical and tropical areas and attacked tropical woody grain crops, such as starchy plants of Artocarpus. Quantitative real-time polymerase chain reaction (qRT-PCR) is a powerful approach for investigating target genes expression profiles at the transcriptional level. However, the identification and selection of internal reference genes, which is often overlooked, is the most vital step before the analysis of target gene expression by qRT-PCR. So far, the reliable internal reference genes under a certain condition of D. caesalis have not been investigated. Therefore, this study evaluated the expression stability of eight candidate reference genes including ACT, β-TUB, GAPDH, G6PDH, RPS3a, RPL13a, EF1α, and EIF4A in different developmental stages, tissues and sexes using geNorm, NormFinder and BestKeeper algorithms. To verify the stability of the recommended internal reference genes, the expression levels of DcaeOBP5 were analyzed under different treatment conditions. The results indicated that ACT, RPL13a, β-TUB, RPS3a, and EF1α were identified as the most stable reference genes for further studies on target gene expression involving different developmental stages of D. caesalis. And ACT and EIF4A were recommended as stable reference genes for different tissues. Furthermore, ACT, EF1α, and RPS3a were ranked as the best reference genes in different sexes based on three algorithms. Our research represents the critical first step to normalize qRT-PCR data and ensure the accuracy of expression of target genes involved in phylogenetic and physiological mechanism at the transcriptional level in D. caesalia.\\n© The Author(s) 2020. Published by Oxford University Press on behalf of Entomological Society of America.\\n',\n",
       "  3),\n",
       " ('\\nThe analysis of inconsistent hydrological series is one of the major problems that should be solved for engineering hydrological calculation in changing environment. In this study, the diffe-rences of non-consistency and non-stationarity were analyzed from the perspective of composition of hydrological series. The inconsistent hydrological phenomena were generalized into hydrological processes with inheritance, variability and evolution characteristics or regulations. Furthermore, the hydrological genes were identified following the theory of biological genes, while their inheritance bases and variability bases were determined based on composition of hydrological series under diffe-rent time scales. To identify and test the components of hydrological genes, we constructed a diagnosis system of hydrological genes. With the P-3 distribution as an example, we described the process of construction and expression of the moment genes to illustrate the inheritance, variability and evolution principles of hydrological genes. With the annual minimum 1-month runoff series of Yunjinghong station in Lancangjiang River basin as an example, we verified the feasibility and practicability of hydrological gene theory for the calculation of inconsistent hydrological frequency. The results showed that the method could be used to reveal the evolution of inconsistent hydrological series. Therefore, it provided a new research pathway for engineering hydrological calculation in changing environment and an essential reference for the assessment of water security.\\n',\n",
       "  2),\n",
       " ('\\nQuantitative real-time PCR (qPCR) is commonly used for deciphering gene functions. For effective qPCR analyses, suitable reference genes are needed for normalization. The objective of this study is to identify the appropriate reference gene(s) for qPCR analyses of the leaves and roots of ramie (Boehmeria nivea L.), an important natural fiber crop. To accomplish this goal, we investigated the expression patterns of eight common plant qPCR reference genes in ramie leaves and roots under five abiotic stresses, five hormonal treatments, and one biotic stress. The relative expression stabilities of the eight genes were evaluated using four common but different approaches: geNorm, NormFinder, BestKeeper, and RefFinder. Across the 11 tested conditions, ACT1 was the most stably expressed among the eight genes while GAPDH displayed the biggest variation. Overall, while variations in the suggested reference genes were found for different tissue x treatment combinations, our analyses revealed that together, genes ACT1, CYP2, and UBQ can provide robust references for gene expression studies of ramie leaves under most conditions, while genes EF-1α, TUB, and ACT1 can be used for similar studies of ramie roots. Our results should help future functional studies of the genes in ramie genome across tissues and environmental conditions.\\n',\n",
       "  2),\n",
       " ('\\nOphraella communa is an effective bio-control agent of the invasive common weed. By now, the reference genes in O. communa have not yet been screened and validated. The aim of this study was to screen for the most stable reference genes in different backgrounds, such as different developmental stages, sexes, tissues, and male reproductive system with different body sizes. We selected 12 common housekeeping genes involved in different biological processes, including GAPDH, ACT1, ACT2, ARF1, ARF4, SDH, βTUBC, RPL4, RPL19, RPS18, EF1α, and COX as the candidate reference genes. To analyze the stability of the candidate reference genes, we first used three dedicated algorithms, GeNorm, NormFinder, and BestKeeper, and further comprehensive ranking was provided by ReFinder. The results showed that RPL19 and RPL4 exhibited the least variation in different developmental stages/sexes and in male reproductive systems with different body sizes. COX proved to be most suitable for normalizing the gene expression levels in different tissues, and coincidentally, RPL19 was also found to be second in terms of stability in this study. To the best of our knowledge, this is the first study to identify suitable reference genes for analyzing gene expression in O. communa; thus, this study would lay the foundation for future research on the molecular physiology and biochemistry of O. communa and other insects.\\nCopyright © 2020 Zhang, Chen, Chen, Ma, Chen, Gao, Tian, Cui, Tian, Guo, Wan and Zhou.\\n',\n",
       "  2)]"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c.execute('''SELECT DISTINCT abstract\n",
    "             FROM PUBLICATION''').fetchall()\n",
    "\n",
    "c.execute('''SELECT auteur_id,\n",
    "             COUNT(*)\n",
    "             FROM PUBLICATION\n",
    "             GROUP BY auteur_id;''').fetchall()\n",
    "\n",
    "c.execute('''SELECT abstract,\n",
    "             COUNT(*)\n",
    "             FROM PUBLICATION\n",
    "             GROUP BY auteur_id\n",
    "             HAVING COUNT(*) >= 2;''').fetchall()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pour faire apparaitre tous les abstracts qui sont supperieur à 1\n",
    "c.execute('''Select abstract from PUBLICATION join (SELECT abstract,\n",
    "             COUNT(*)\n",
    "             FROM PUBLICATION\n",
    "             GROUP BY auteur_id\n",
    "             HAVING COUNT(*) >= 2;) \n",
    "             AS TWO_AUTHORS ON PUBLICATION.auteur_id = TWO_AUTHORS.auteur_id;''').fetchall()\n",
    "#Shady: SELECT abstract FROM pubmed\n",
    "#JOIN (SELECT last_author FROM pubmed GROUP BY last_author HAVING COUNT(last_author) > 1)\n",
    "#AS two_authors ON pubmed.last_author=two_authors.last_author"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T14:29:32.183042Z",
     "start_time": "2020-06-24T14:29:32.054607Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('Gene, Gene', 102),\n",
       " ('Genes, Genes (Basel)', 75),\n",
       " ('PloS one, PLoS ONE', 75),\n",
       " ('Scientific reports, Sci Rep', 44),\n",
       " ('BMC genomics, BMC Genomics', 30),\n",
       " ('Molecular medicine reports, Mol Med Rep', 22),\n",
       " ('BioMed research international, Biomed Res Int', 14),\n",
       " ('Frontiers in genetics, Front Genet', 14),\n",
       " ('Molecular biology reports, Mol. Biol. Rep.', 13),\n",
       " ('PeerJ, PeerJ', 13),\n",
       " ('BMC plant biology, BMC Plant Biol.', 12),\n",
       " ('BMC bioinformatics, BMC Bioinformatics', 10),\n",
       " ('Oncology letters, Oncol Lett', 10),\n",
       " ('BMC medical genomics, BMC Med Genomics', 9),\n",
       " ('Frontiers in plant science, Front Plant Sci', 9),\n",
       " ('Medical science monitor : international medical journal of experimental and clinical research, Med. Sci. Monit.',\n",
       "  9),\n",
       " ('Molecular biology and evolution, Mol. Biol. Evol.', 8),\n",
       " ('International journal of molecular sciences, Int J Mol Sci', 7),\n",
       " ('Journal of microbiological methods, J. Microbiol. Methods', 7),\n",
       " ('Molecular genetics and genomics : MGG, Mol. Genet. Genomics', 7),\n",
       " ('Proceedings of the National Academy of Sciences of the United States of America, Proc. Natl. Acad. Sci. U.S.A.',\n",
       "  7),\n",
       " ('Nucleic acids research, Nucleic Acids Res.', 6),\n",
       " ('Plant physiology and biochemistry : PPB, Plant Physiol. Biochem.', 6),\n",
       " ('BMC research notes, BMC Res Notes', 5),\n",
       " ('Frontiers in microbiology, Front Microbiol', 5),\n",
       " ('G3 (Bethesda, Md.), G3 (Bethesda)', 5),\n",
       " ('Genome biology and evolution, Genome Biol Evol', 5),\n",
       " ('Journal of cellular biochemistry, J. Cell. Biochem.', 5),\n",
       " ('Oncotarget, Oncotarget', 5),\n",
       " ('American journal of human genetics, Am. J. Hum. Genet.', 4),\n",
       " ('Applied and environmental microbiology, Appl. Environ. Microbiol.', 4),\n",
       " ('BMC evolutionary biology, BMC Evol. Biol.', 4),\n",
       " ('Frontiers in physiology, Front Physiol', 4),\n",
       " ('Journal of experimental botany, J. Exp. Bot.', 4),\n",
       " ('Medicine, Medicine (Baltimore)', 4),\n",
       " ('Molecules (Basel, Switzerland), Molecules', 4),\n",
       " ('Plant biotechnology journal, Plant Biotechnol. J.', 4),\n",
       " ('TAG. Theoretical and applied genetics. Theoretische und angewandte Genetik, Theor. Appl. Genet.',\n",
       "  4),\n",
       " ('The New phytologist, New Phytol.', 4),\n",
       " ('American journal of medical genetics. Part B, Neuropsychiatric genetics : the official publication of the International Society of Psychiatric Genetics, Am. J. Med. Genet. B Neuropsychiatr. Genet.',\n",
       "  3),\n",
       " ('Bioinformatics (Oxford, England), Bioinformatics', 3),\n",
       " ('Cancer management and research, Cancer Manag Res', 3),\n",
       " ('Current genetics, Curr. Genet.', 3),\n",
       " ('Epigenetics & chromatin, Epigenetics Chromatin', 3),\n",
       " ('Experimental and therapeutic medicine, Exp Ther Med', 3),\n",
       " ('Frontiers in oncology, Front Oncol', 3),\n",
       " ('Genes & genomics, Genes Genomics', 3),\n",
       " ('Genome research, Genome Res.', 3),\n",
       " ('Human molecular genetics, Hum. Mol. Genet.', 3),\n",
       " ('Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases, Infect. Genet. Evol.',\n",
       "  3),\n",
       " ('International journal of cancer, Int. J. Cancer', 3),\n",
       " ('Journal of genetics, J. Genet.', 3),\n",
       " ('Journal of insect science (Online), J. Insect Sci.', 3),\n",
       " ('Journal of theoretical biology, J. Theor. Biol.', 3),\n",
       " ('Letters in applied microbiology, Lett. Appl. Microbiol.', 3),\n",
       " ('Methods in molecular biology (Clifton, N.J.), Methods Mol. Biol.', 3),\n",
       " ('Molecular plant pathology, Mol. Plant Pathol.', 3),\n",
       " ('Oncology reports, Oncol. Rep.', 3),\n",
       " ('Physiology and molecular biology of plants : an international journal of functional plant biology, Physiol Mol Biol Plants',\n",
       "  3),\n",
       " ('Plant molecular biology, Plant Mol. Biol.', 3),\n",
       " ('The Plant journal : for cell and molecular biology, Plant J.', 3),\n",
       " ('Aging, Aging (Albany NY)', 2),\n",
       " ('Applied microbiology and biotechnology, Appl. Microbiol. Biotechnol.', 2),\n",
       " ('Aquatic toxicology (Amsterdam, Netherlands), Aquat. Toxicol.', 2),\n",
       " ('BMC biotechnology, BMC Biotechnol.', 2),\n",
       " ('BMC cancer, BMC Cancer', 2),\n",
       " ('BMC genetics, BMC Genet.', 2),\n",
       " ('Bio Systems, BioSystems', 2),\n",
       " ('Biomedical reports, Biomed Rep', 2),\n",
       " ('Biomedicine & pharmacotherapy = Biomedecine & pharmacotherapie, Biomed. Pharmacother.',\n",
       "  2),\n",
       " ('Cell cycle (Georgetown, Tex.), Cell Cycle', 2),\n",
       " ('Cell reports, Cell Rep', 2),\n",
       " ('Cells, Cells', 2),\n",
       " ('Critical reviews in eukaryotic gene expression, Crit. Rev. Eukaryot. Gene Expr.',\n",
       "  2),\n",
       " ('Cytokine, Cytokine', 2),\n",
       " ('DNA research : an international journal for rapid publication of reports on genes and genomes, DNA Res.',\n",
       "  2),\n",
       " ('Data in brief, Data Brief', 2),\n",
       " ('Developmental biology, Dev. Biol.', 2),\n",
       " ('Endocrine, Endocrine', 2),\n",
       " ('FEBS open bio, FEBS Open Bio', 2),\n",
       " ('Fish physiology and biochemistry, Fish Physiol. Biochem.', 2),\n",
       " ('Fungal genetics and biology : FG & B, Fungal Genet. Biol.', 2),\n",
       " ('Genetics, Genetics', 2),\n",
       " ('Genome biology, Genome Biol.', 2),\n",
       " ('Hepatology research : the official journal of the Japan Society of Hepatology, Hepatol. Res.',\n",
       "  2),\n",
       " ('Hereditas, Hereditas', 2),\n",
       " ('IEEE/ACM transactions on computational biology and bioinformatics, IEEE/ACM Trans Comput Biol Bioinform',\n",
       "  2),\n",
       " ('Insect biochemistry and molecular biology, Insect Biochem. Mol. Biol.', 2),\n",
       " ('Interdisciplinary sciences, computational life sciences, Interdiscip Sci',\n",
       "  2),\n",
       " ('Journal of Cancer, J Cancer', 2),\n",
       " ('Journal of applied genetics, J. Appl. Genet.', 2),\n",
       " ('Journal of cellular physiology, J. Cell. Physiol.', 2),\n",
       " ('Journal of computational biology : a journal of computational molecular cell biology, J. Comput. Biol.',\n",
       "  2),\n",
       " ('Journal of economic entomology, J. Econ. Entomol.', 2),\n",
       " ('Journal of immunology (Baltimore, Md. : 1950), J. Immunol.', 2),\n",
       " ('Journal of molecular evolution, J. Mol. Evol.', 2),\n",
       " ('Journal of molecular neuroscience : MN, J. Mol. Neurosci.', 2),\n",
       " ('Journal of thoracic disease, J Thorac Dis', 2),\n",
       " ('Journal of virology, J. Virol.', 2),\n",
       " ('Mammalian genome : official journal of the International Mammalian Genome Society, Mamm. Genome',\n",
       "  2),\n",
       " ('Melanoma research, Melanoma Res.', 2),\n",
       " ('Microbiology (Reading, England), Microbiology (Reading, Engl.)', 2),\n",
       " ('Molecular and cellular biochemistry, Mol. Cell. Biochem.', 2),\n",
       " ('Molecular genetics & genomic medicine, Mol Genet Genomic Med', 2),\n",
       " ('Molecular microbiology, Mol. Microbiol.', 2),\n",
       " ('Molecular phylogenetics and evolution, Mol. Phylogenet. Evol.', 2),\n",
       " ('Molekuliarnaia biologiia, Mol. Biol. (Mosk.)', 2),\n",
       " ('Neurological research, Neurol. Res.', 2),\n",
       " ('PLoS genetics, PLoS Genet.', 2),\n",
       " ('Pesticide biochemistry and physiology, Pestic Biochem Physiol', 2),\n",
       " ('Physiological genomics, Physiol. Genomics', 2),\n",
       " ('Plant & cell physiology, Plant Cell Physiol.', 2),\n",
       " ('Plant cell reports, Plant Cell Rep.', 2),\n",
       " ('Plant physiology, Plant Physiol.', 2),\n",
       " ('Planta, Planta', 2),\n",
       " ('The Plant cell, Plant Cell', 2),\n",
       " ('Toxins, Toxins (Basel)', 2),\n",
       " ('Tree physiology, Tree Physiol.', 2),\n",
       " ('World journal of microbiology & biotechnology, World J. Microbiol. Biotechnol.',\n",
       "  2),\n",
       " ('World journal of surgical oncology, World J Surg Oncol', 2),\n",
       " ('Ying yong sheng tai xue bao = The journal of applied ecology, Ying Yong Sheng Tai Xue Bao',\n",
       "  2),\n",
       " ('3 Biotech, 3 Biotech', 1),\n",
       " ('Acta parasitologica, Acta Parasitol.', 1),\n",
       " ('Acta reumatologica portuguesa, Acta Reumatol Port', 1),\n",
       " ('Acta veterinaria Hungarica, Acta Vet. Hung.', 1),\n",
       " ('Advances in clinical and experimental medicine : official organ Wroclaw Medical University, Adv Clin Exp Med',\n",
       "  1),\n",
       " ('Aging clinical and experimental research, Aging Clin Exp Res', 1),\n",
       " ('Anaerobe, Anaerobe', 1),\n",
       " ('Andrologia, Andrologia', 1),\n",
       " ('Animal : an international journal of animal bioscience, Animal', 1),\n",
       " ('Annals of anatomy = Anatomischer Anzeiger : official organ of the Anatomische Gesellschaft, Ann. Anat.',\n",
       "  1),\n",
       " ('Annals of translational medicine, Ann Transl Med', 1),\n",
       " ('Antonie van Leeuwenhoek, Antonie Van Leeuwenhoek', 1),\n",
       " ('Applied biochemistry and biotechnology, Appl. Biochem. Biotechnol.', 1),\n",
       " ('Archives of dermatological research, Arch. Dermatol. Res.', 1),\n",
       " ('Archives of gynecology and obstetrics, Arch. Gynecol. Obstet.', 1),\n",
       " ('Archives of insect biochemistry and physiology, Arch. Insect Biochem. Physiol.',\n",
       "  1),\n",
       " ('Archives of oral biology, Arch. Oral Biol.', 1),\n",
       " ('Archives of virology, Arch. Virol.', 1),\n",
       " ('Asia-Pacific journal of ophthalmology (Philadelphia, Pa.), Asia Pac J Ophthalmol (Phila)',\n",
       "  1),\n",
       " ('Asia-Pacific psychiatry : official journal of the Pacific Rim College of Psychiatrists, Asia Pac Psychiatry',\n",
       "  1),\n",
       " ('BMB reports, BMB Rep', 1),\n",
       " ('BMC biology, BMC Biol.', 1),\n",
       " ('BMC cardiovascular disorders, BMC Cardiovasc Disord', 1),\n",
       " ('BMC immunology, BMC Immunol.', 1),\n",
       " ('BMC medical ethics, BMC Med Ethics', 1),\n",
       " ('BMC medical genetics, BMC Med. Genet.', 1),\n",
       " ('BMC microbiology, BMC Microbiol.', 1),\n",
       " ('BMC molecular biology, BMC Mol. Biol.', 1),\n",
       " ('BMC neuroscience, BMC Neurosci', 1),\n",
       " ('BMC systems biology, BMC Syst Biol', 1),\n",
       " ('Behavioural brain research, Behav. Brain Res.', 1),\n",
       " ('Beijing da xue xue bao. Yi xue ban = Journal of Peking University. Health sciences, Beijing Da Xue Xue Bao',\n",
       "  1),\n",
       " ('Biochemical genetics, Biochem. Genet.', 1),\n",
       " ('Biochimica et biophysica acta. Molecular basis of disease, Biochim Biophys Acta Mol Basis Dis',\n",
       "  1),\n",
       " ('Biological research, Biol. Res.', 1),\n",
       " ('Biological trace element research, Biol Trace Elem Res', 1),\n",
       " ('Biology methods & protocols, Biol Methods Protoc', 1),\n",
       " ('Biometals : an international journal on the role of metal ions in biology, biochemistry, and medicine, Biometals',\n",
       "  1),\n",
       " ('Biomolecules, Biomolecules', 1),\n",
       " ('Bioresource technology, Bioresour. Technol.', 1),\n",
       " ('Bioscience reports, Biosci. Rep.', 1),\n",
       " ('Biotechnology and applied biochemistry, Biotechnol. Appl. Biochem.', 1),\n",
       " ('Biotechnology for biofuels, Biotechnol Biofuels', 1),\n",
       " ('Biotechnology letters, Biotechnol. Lett.', 1),\n",
       " ('Bone, Bone', 1),\n",
       " ('Brain research bulletin, Brain Res. Bull.', 1),\n",
       " ('Brazilian journal of biology = Revista brasleira de biologia, Braz J Biol',\n",
       "  1),\n",
       " ('Breast cancer research and treatment, Breast Cancer Res. Treat.', 1),\n",
       " ('Briefings in functional genomics, Brief Funct Genomics', 1),\n",
       " ('Bulletin of entomological research, Bull. Entomol. Res.', 1),\n",
       " ('Canadian journal of microbiology, Can. J. Microbiol.', 1),\n",
       " ('Cancer biology & therapy, Cancer Biol. Ther.', 1),\n",
       " ('Cancer biomarkers : section A of Disease markers, Cancer Biomark', 1),\n",
       " ('Cancer gene therapy, Cancer Gene Ther.', 1),\n",
       " ('Cancer genetics, Cancer Genet', 1),\n",
       " ('Cancer informatics, Cancer Inform', 1),\n",
       " ('Cancer investigation, Cancer Invest.', 1),\n",
       " ('Cancer medicine, Cancer Med', 1),\n",
       " ('Cardiovascular research, Cardiovasc. Res.', 1),\n",
       " ('Cell biochemistry and biophysics, Cell Biochem. Biophys.', 1),\n",
       " ('Cell journal, Cell J', 1),\n",
       " ('Cellular oncology (Dordrecht), Cell Oncol (Dordr)', 1),\n",
       " ('Cellular physiology and biochemistry : international journal of experimental cellular physiology, biochemistry, and pharmacology, Cell. Physiol. Biochem.',\n",
       "  1),\n",
       " ('Chromosoma, Chromosoma', 1),\n",
       " ('Chronobiology international, Chronobiol. Int.', 1),\n",
       " ('Circulation, Circulation', 1),\n",
       " ('Clinical and translational science, Clin Transl Sci', 1),\n",
       " ('Computational and mathematical methods in medicine, Comput Math Methods Med',\n",
       "  1),\n",
       " ('Computational and structural biotechnology journal, Comput Struct Biotechnol J',\n",
       "  1),\n",
       " ('Computational biology and chemistry, Comput Biol Chem', 1),\n",
       " ('Computers in biology and medicine, Comput. Biol. Med.', 1),\n",
       " ('Current molecular pharmacology, Curr Mol Pharmacol', 1),\n",
       " ('Discovery medicine, Discov Med', 1),\n",
       " ('Disease models & mechanisms, Dis Model Mech', 1),\n",
       " ('Ecotoxicology and environmental safety, Ecotoxicol. Environ. Saf.', 1),\n",
       " ('Endocrine journal, Endocr. J.', 1),\n",
       " ('Environmental microbiology, Environ. Microbiol.', 1),\n",
       " ('Environmental pollution (Barking, Essex : 1987), Environ. Pollut.', 1),\n",
       " ('Environmental science & technology, Environ. Sci. Technol.', 1),\n",
       " ('Environmental toxicology and pharmacology, Environ. Toxicol. Pharmacol.',\n",
       "  1),\n",
       " ('Enzyme and microbial technology, Enzyme Microb. Technol.', 1),\n",
       " ('Ethiopian journal of health sciences, Ethiop J Health Sci', 1),\n",
       " ('Eukaryotic cell, Eukaryotic Cell', 1),\n",
       " ('European journal of cancer prevention : the official journal of the European Cancer Prevention Organisation (ECP), Eur. J. Cancer Prev.',\n",
       "  1),\n",
       " ('Experimental physiology, Exp. Physiol.', 1),\n",
       " ('FEMS microbiology letters, FEMS Microbiol. Lett.', 1),\n",
       " ('Fish & shellfish immunology, Fish Shellfish Immunol.', 1),\n",
       " ('Frontiers in bioengineering and biotechnology, Front Bioeng Biotechnol', 1),\n",
       " ('Frontiers in immunology, Front Immunol', 1),\n",
       " ('Frontiers in medicine, Front Med (Lausanne)', 1),\n",
       " ('Frontiers in systems neuroscience, Front Syst Neurosci', 1),\n",
       " ('Functional & integrative genomics, Funct. Integr. Genomics', 1),\n",
       " ('Gene expression patterns : GEP, Gene Expr. Patterns', 1),\n",
       " ('Gene reports, Gene Rep', 1),\n",
       " ('Genes & genetic systems, Genes Genet. Syst.', 1),\n",
       " ('Genes and immunity, Genes Immun.', 1),\n",
       " ('Genetic epidemiology, Genet. Epidemiol.', 1),\n",
       " ('Genetic testing and molecular biomarkers, Genet Test Mol Biomarkers', 1),\n",
       " ('Genetics and molecular research : GMR, Genet. Mol. Res.', 1),\n",
       " ('Genetics, selection, evolution : GSE, Genet. Sel. Evol.', 1),\n",
       " ('Genome medicine, Genome Med', 1),\n",
       " ('Genome, Genome', 1),\n",
       " ('GigaScience, Gigascience', 1),\n",
       " ('Glycobiology, Glycobiology', 1),\n",
       " ('Hematology (Amsterdam, Netherlands), Hematology', 1),\n",
       " ('Histopathology, Histopathology', 1),\n",
       " ('Huan jing ke xue= Huanjing kexue, Huan Jing Ke Xue', 1),\n",
       " ('Human genetics, Hum. Genet.', 1),\n",
       " ('Human mutation, Hum. Mutat.', 1),\n",
       " ('IET systems biology, IET Syst Biol', 1),\n",
       " ('Inflammatory bowel diseases, Inflamm. Bowel Dis.', 1),\n",
       " ('Insect molecular biology, Insect Mol. Biol.', 1),\n",
       " ('Insects, Insects', 1),\n",
       " ('International journal for parasitology, Int. J. Parasitol.', 1),\n",
       " ('International journal of antimicrobial agents, Int. J. Antimicrob. Agents',\n",
       "  1),\n",
       " ('International journal of food microbiology, Int. J. Food Microbiol.', 1),\n",
       " ('International journal of genomics, Int J Genomics', 1),\n",
       " ('International journal of immunogenetics, Int. J. Immunogenet.', 1),\n",
       " ('International journal of legal medicine, Int. J. Legal Med.', 1),\n",
       " ('International journal of medicinal mushrooms, Int J Med Mushrooms', 1),\n",
       " ('International journal of molecular epidemiology and genetics, Int J Mol Epidemiol Genet',\n",
       "  1),\n",
       " ('International journal of molecular medicine, Int. J. Mol. Med.', 1),\n",
       " ('International journal of obesity (2005), Int J Obes (Lond)', 1),\n",
       " ('International journal of ophthalmology, Int J Ophthalmol', 1),\n",
       " ('International journal of rheumatic diseases, Int J Rheum Dis', 1),\n",
       " ('International journal of systematic and evolutionary microbiology, Int. J. Syst. Evol. Microbiol.',\n",
       "  1),\n",
       " ('Iranian journal of basic medical sciences, Iran J Basic Med Sci', 1),\n",
       " ('Iranian journal of pathology, Iran J Pathol', 1),\n",
       " (\"Journal of Alzheimer's disease : JAD, J. Alzheimers Dis.\", 1),\n",
       " ('Journal of affective disorders, J Affect Disord', 1),\n",
       " ('Journal of agricultural and food chemistry, J. Agric. Food Chem.', 1),\n",
       " ('Journal of applied microbiology, J. Appl. Microbiol.', 1),\n",
       " ('Journal of bioinformatics and computational biology, J Bioinform Comput Biol',\n",
       "  1),\n",
       " ('Journal of biomedical informatics, J Biomed Inform', 1),\n",
       " ('Journal of biomedical semantics, J Biomed Semantics', 1),\n",
       " ('Journal of bioscience and bioengineering, J. Biosci. Bioeng.', 1),\n",
       " ('Journal of biotechnology, J. Biotechnol.', 1),\n",
       " ('Journal of cancer research and clinical oncology, J. Cancer Res. Clin. Oncol.',\n",
       "  1),\n",
       " ('Journal of cardiovascular translational research, J Cardiovasc Transl Res',\n",
       "  1),\n",
       " ('Journal of cellular and molecular medicine, J. Cell. Mol. Med.', 1),\n",
       " ('Journal of child psychology and psychiatry, and allied disciplines, J Child Psychol Psychiatry',\n",
       "  1),\n",
       " ('Journal of clinical neurology (Seoul, Korea), J Clin Neurol', 1),\n",
       " ('Journal of environmental science and health. Part. B, Pesticides, food contaminants, and agricultural wastes, J Environ Sci Health B',\n",
       "  1),\n",
       " ('Journal of fish biology, J. Fish Biol.', 1),\n",
       " ('Journal of global antimicrobial resistance, J Glob Antimicrob Resist', 1),\n",
       " ('Journal of hazardous materials, J. Hazard. Mater.', 1),\n",
       " ('Journal of immunological methods, J. Immunol. Methods', 1),\n",
       " ('Journal of immunology research, J Immunol Res', 1),\n",
       " ('Journal of insect physiology, J. Insect Physiol.', 1),\n",
       " ('Journal of invertebrate pathology, J. Invertebr. Pathol.', 1),\n",
       " ('Journal of neuroscience research, J. Neurosci. Res.', 1),\n",
       " ('Journal of orthopaedic surgery and research, J Orthop Surg Res', 1),\n",
       " ('Journal of plant research, J. Plant Res.', 1),\n",
       " ('Journal of psychopharmacology (Oxford, England), J. Psychopharmacol. (Oxford)',\n",
       "  1),\n",
       " ('Journal of the science of food and agriculture, J. Sci. Food Agric.', 1),\n",
       " ('Journal of translational medicine, J Transl Med', 1),\n",
       " ('Lab on a chip, Lab Chip', 1),\n",
       " ('Le infezioni in medicina, Infez Med', 1),\n",
       " ('Lipids, Lipids', 1),\n",
       " ('Marine biotechnology (New York, N.Y.), Mar. Biotechnol.', 1),\n",
       " ('Mathematical biosciences, Math Biosci', 1),\n",
       " ('Medicinal chemistry (Shariqah (United Arab Emirates)), Med Chem', 1),\n",
       " ('Methods (San Diego, Calif.), Methods', 1),\n",
       " ('Microbial cell factories, Microb. Cell Fact.', 1),\n",
       " ('Microbial ecology, Microb. Ecol.', 1),\n",
       " ('Microbial pathogenesis, Microb. Pathog.', 1),\n",
       " ('Microbiological research, Microbiol. Res.', 1),\n",
       " ('Microbiome, Microbiome', 1),\n",
       " ('Microorganisms, Microorganisms', 1),\n",
       " ('Molecular and cellular neurosciences, Mol. Cell. Neurosci.', 1),\n",
       " ('Molecular and cellular probes, Mol. Cell. Probes', 1),\n",
       " ('Molecular bioSystems, Mol Biosyst', 1),\n",
       " ('Molecular ecology, Mol. Ecol.', 1),\n",
       " ('Molecular human reproduction, Mol. Hum. Reprod.', 1),\n",
       " ('Molecular imaging and biology, Mol Imaging Biol', 1),\n",
       " ('Molecular neurobiology, Mol. Neurobiol.', 1),\n",
       " ('Molecular oncology, Mol Oncol', 1),\n",
       " ('NPJ genomic medicine, NPJ Genom Med', 1),\n",
       " ('Nature genetics, Nat. Genet.', 1),\n",
       " ('Nephrology, dialysis, transplantation : official publication of the European Dialysis and Transplant Association - European Renal Association, Nephrol. Dial. Transplant.',\n",
       "  1),\n",
       " ('Neural development, Neural Dev', 1),\n",
       " ('Neurogenetics, Neurogenetics', 1),\n",
       " ('Neuropsychiatric disease and treatment, Neuropsychiatr Dis Treat', 1),\n",
       " ('Nitric oxide : biology and chemistry, Nitric Oxide', 1),\n",
       " ('Nutrients, Nutrients', 1),\n",
       " ('OncoTargets and therapy, Onco Targets Ther', 1),\n",
       " ('Oncogene, Oncogene', 1),\n",
       " ('Pathologie-biologie, Pathol. Biol.', 1),\n",
       " ('Pathology oncology research : POR, Pathol. Oncol. Res.', 1),\n",
       " ('Pathology, research and practice, Pathol. Res. Pract.', 1),\n",
       " ('Pediatric research, Pediatr. Res.', 1),\n",
       " ('Pest management science, Pest Manag. Sci.', 1),\n",
       " ('Philosophical transactions of the Royal Society of London. Series B, Biological sciences, Philos. Trans. R. Soc. Lond., B, Biol. Sci.',\n",
       "  1),\n",
       " ('Physiological research, Physiol Res', 1),\n",
       " ('Phytopathology, Phytopathology', 1),\n",
       " ('Plant biology (Stuttgart, Germany), Plant Biol (Stuttg)', 1),\n",
       " ('Plant disease, Plant Dis.', 1),\n",
       " ('Plant methods, Plant Methods', 1),\n",
       " ('Plant, cell & environment, Plant Cell Environ.', 1),\n",
       " ('Plastic and reconstructive surgery, Plast. Reconstr. Surg.', 1),\n",
       " ('Polish journal of veterinary sciences, Pol J Vet Sci', 1),\n",
       " ('Poultry science, Poult. Sci.', 1),\n",
       " ('Progress in neuro-psychopharmacology & biological psychiatry, Prog. Neuropsychopharmacol. Biol. Psychiatry',\n",
       "  1),\n",
       " ('Psychopharmacology, Psychopharmacology (Berl.)', 1),\n",
       " ('Radiation and environmental biophysics, Radiat Environ Biophys', 1),\n",
       " ('Radiation research, Radiat. Res.', 1),\n",
       " ('Research in microbiology, Res. Microbiol.', 1),\n",
       " ('Rice (New York, N.Y.), Rice (N Y)', 1),\n",
       " ('Surgical oncology, Surg Oncol', 1),\n",
       " ('Technology in cancer research & treatment, Technol. Cancer Res. Treat.', 1),\n",
       " ('The Journal of allergy and clinical immunology, J. Allergy Clin. Immunol.',\n",
       "  1),\n",
       " ('The Journal of biological chemistry, J. Biol. Chem.', 1),\n",
       " ('The Journal of general and applied microbiology, J. Gen. Appl. Microbiol.',\n",
       "  1),\n",
       " ('The Journal of international medical research, J. Int. Med. Res.', 1),\n",
       " ('The Science of the total environment, Sci. Total Environ.', 1),\n",
       " ('The journals of gerontology. Series A, Biological sciences and medical sciences, J. Gerontol. A Biol. Sci. Med. Sci.',\n",
       "  1),\n",
       " ('The world journal of biological psychiatry : the official journal of the World Federation of Societies of Biological Psychiatry, World J. Biol. Psychiatry',\n",
       "  1),\n",
       " ('Theriogenology, Theriogenology', 1),\n",
       " ('Toxicology and industrial health, Toxicol Ind Health', 1),\n",
       " ('Transcription, Transcription', 1),\n",
       " ('Translational psychiatry, Transl Psychiatry', 1),\n",
       " ('Transplantation proceedings, Transplant. Proc.', 1),\n",
       " ('Tumori, Tumori', 1),\n",
       " ('Twin research and human genetics : the official journal of the International Society for Twin Studies, Twin Res Hum Genet',\n",
       "  1),\n",
       " ('Vascular, Vascular', 1),\n",
       " ('Veterinary immunology and immunopathology, Vet. Immunol. Immunopathol.', 1),\n",
       " ('Veterinary microbiology, Vet. Microbiol.', 1),\n",
       " ('Virology, Virology', 1),\n",
       " ('Virulence, Virulence', 1),\n",
       " ('Viruses, Viruses', 1),\n",
       " ('Water environment research : a research publication of the Water Environment Federation, Water Environ. Res.',\n",
       "  1),\n",
       " ('World journal of gastroenterology, World J. Gastroenterol.', 1),\n",
       " ('Yonsei medical journal, Yonsei Med. J.', 1),\n",
       " ('Zeitschrift fur Naturforschung. C, Journal of biosciences, Z. Naturforsch., C, J. Biosci.',\n",
       "  1),\n",
       " ('Zoological science, Zool. Sci.', 1)]"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c.execute('''SELECT journal, \n",
    "             COUNT(*)\n",
    "             FROM JOURNAL_TABLE\n",
    "             INNER JOIN PUBLICATION ON PUBLICATION.journal_id = JOURNAL_TABLE.journal_id\n",
    "             GROUP BY PUBLICATION.journal_id\n",
    "             ORDER BY COUNT(*) DESC;''').fetchall()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-24T14:50:17.559357Z",
     "start_time": "2020-06-24T14:50:17.531275Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('2010', 2),\n",
       " ('2011', 3),\n",
       " ('2012', 7),\n",
       " ('2013', 27),\n",
       " ('2014', 48),\n",
       " ('2015', 81),\n",
       " ('2016', 87),\n",
       " ('2017', 151),\n",
       " ('2018', 192),\n",
       " ('2019', 297),\n",
       " ('2020', 105)]"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# http://aprogrammersday.blogspot.com/2014/02/group-by-month-or-year-and-sum-in-sqlite.html\n",
    "# https://kimsereyblog.blogspot.com/2016/06/nice-features-and-tricks-with-sqlite.html\n",
    "\n",
    "c.execute('''SELECT strftime('%Y', publication_date) as valYear, \n",
    "             COUNT(*) as valTotalYear \n",
    "             FROM PUBLICATION\n",
    "             GROUP BY valYear;''').fetchall()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# or pour le nbr d'articles par an\n",
    "SELECT COUNT(title), strftime('%Y', publication_date) FROM PUBLICATION GROUP BY strftime('%Y', publication_date)\n",
    "\n",
    "# Shady: SELECT COUNT(title), strftime('%Y', publication_date) FROM pubmed GROUP BY strftime('%Y', publication_date)"
   ]
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
